<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>QC with STAR and Qualimap | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="QC with STAR and Qualimap" />
<meta name="author" content="Meeta Mistry, Mary Piper, Radhika Khetani, Jihe Liu - Edited by Sally Chang @ NICHD" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/10_QC_Qualimap.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/10_QC_Qualimap.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2024-10-07T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="QC with STAR and Qualimap" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Meeta Mistry, Mary Piper, Radhika Khetani, Jihe Liu - Edited by Sally Chang @ NICHD"},"dateModified":"2024-10-07T00:00:00+00:00","datePublished":"2024-10-07T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"QC with STAR and Qualimap","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/10_QC_Qualimap.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/10_QC_Qualimap.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/modified-intro-march2025/assets/css/style.css?v=b507c6c38fbc65ecc2e6abea386dbda0aeab8b3b">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/modified-intro-march2025/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">QC with STAR and Qualimap</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <p>Approximate time: 50 minutes</p>

<h2 id="learning-objectives">Learning Objectives:</h2>

<ul>
  <li>Run an alignment tool to generate BAM files for input to Qualimap</li>
  <li>Run Qualimap to compute metrics on alignment files</li>
</ul>

<h2 id="quality-control">Quality Control</h2>

<p>After running Salmon, we now have transcript-level abundance estimates for each of our samples, however we don’t have a good way to be able to determine the following:</p>

<ul>
  <li>
    <p>Are there more reads from one end of the gene or another? I.e. do we have 3’ or 5’ bias.</p>
  </li>
  <li>
    <p>What proportion of total reads are mapping to exonic regions? If there are a lot of intergenic or intronic reads, it may indicate DNA contamination.</p>
  </li>
</ul>

<p><strong>In order for us to assess these metrics we need the genomic coordinates of where each read maps</strong>. This information can be generated by mapping (or aligning) reads to the whole genome using genome alignment tools, and can then be used as input to tools that can generate the desired metrics.</p>

<p><img src="../img/full_workflow_qualimap_2019.png" /></p>

<h2 id="qualimap">Qualimap</h2>

<p>We will be using <a href="http://qualimap.bioinfo.cipf.es/doc_html/intro.html#what-is-qualimap">Qualimap</a>, a Java application, which <strong>provides an overall view of the data quality in an HTML report</strong>. The report is comprehensive with mapping statistics summarized along with useful figures and tab-delimited files of metrics data. We will run Qualimap on a single sample, and then take a detailed look at the report and metrics in comparison to what we would expect for good data.</p>

<p><strong>Before we can run Qualimap, we need to use a genome alignment tool to get the genomic coordinates of where the reads are aligning to the genome</strong>. Most aligners that work with NGS data generate a very specific file format wherein all the alignment results are organized systematically. This file format is called SAM/BAM (see below).</p>

<blockquote>
  <h4 id="alignment-file-format-sambam">Alignment file format: SAM/BAM</h4>

  <p>The <strong>Sequence Alignment Map format</strong> (SAM) file is <strong>a tab-delimited text file that contains all information from the FASTQ file, with additional fields containing alignment information for each read</strong>. Specifically, we can obtain the genomic coordinates of where each read maps to in the genome and the quality of that mapping. A <strong>BAM file is the binary, compressed version of the SAM file</strong>. It is significantly smaller in size and is usually the file format requested for by downstream tools that require alignment data as input. The paper by <a href="http://bioinformatics.oxfordjournals.org/content/25/16/2078.full">Heng Li et al</a> provides a lot more detail on the specification.</p>
</blockquote>

<h3 id="creating-a-bam-file">Creating a BAM file</h3>

<p>To generate a BAM file we need to <strong>map our reads to the genome</strong>. The alignment process consists of choosing an appropriate reference genome to map our reads against and performing the read alignment using one of several splice-aware alignment tools such as <a href="http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635">STAR</a> or <a href="http://ccb.jhu.edu/software/hisat2/index.shtml">HISAT2</a>. The choice of aligner is often a personal preference and also dependent on the computational resources that are available to you.</p>

<h3 id="star-aligner">STAR Aligner</h3>

<p>To determine where on the human genome our reads originated from, we will align our reads to the reference genome using <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/">STAR</a> (Spliced Transcripts Alignment to a Reference). STAR is an aligner designed to specifically address many of the challenges of RNA-seq data mapping using a strategy to account for spliced alignments. We will not go into detail about how STAR works, but if you are interested in understanding the alignment strategy we have <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/lessons/STAR_alignment_strategy.html">some materials linked here</a>.</p>

<blockquote>
  <p><strong>NOTE</strong>: Until recently, the standard approach for RNA-seq analysis has been to map our reads using a splice-aware aligner (i.e STAR) and then use the resulting BAM files as input to counting tools like <a href="http://subread.sourceforge.net/">featureCounts</a> and <a href="https://htseq.readthedocs.io/en/release_0.11.1/count.html">htseq-count</a> to obtain our final expression matrix. The field has now moved towards using lightweight alignment tools like Salmon as standard practice, and so we only use STAR for generating a BAM file. If you are interested in knowing more about the standard approach we have some <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon/lessons/counting_reads.html">materials linked here</a>.</p>
</blockquote>

<p>To get started with this lesson, start an interactive session with 6 cores and 8G of memory:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>sinteractive <span class="nt">--cpus-per-task</span><span class="o">=</span>6 <span class="nt">--mem</span><span class="o">=</span>8G  
</code></pre></div></div>

<p>You should have a directory tree setup similar to that shown below. It is best practice to have all files you intend on using for your workflow present within the same directory.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>rnaseq
    ├── logs
    ├── meta
    ├── raw_data
    │   ├── Irrel_kd_1.subset.fq
    │   ├── Irrel_kd_2.subset.fq
    │   ├── Irrel_kd_3.subset.fq
    │   ├── Mov10_oe_1.subset.fq
    │   ├── Mov10_oe_2.subset.fq
    │   └── Mov10_oe_3.subset.fq
    ├── results
    └── scripts
</code></pre></div></div>

<p>To use the STAR aligner, load the module:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>module load star/2.7.6a <span class="c">#needed newer version of STAR than what was listed in tutorial originally</span>
</code></pre></div></div>

<p>Similar to Salmon, aligning reads using STAR is <strong>a two step process</strong>:</p>

<ol>
  <li>Create a genome index</li>
  <li>Map reads to the genome</li>
</ol>

<blockquote>
  <p><strong><em>UPDATE BASED ON SHARED LOCATION FOR BSPC AND/OR BIOWULF</em></strong></p>

  <p>A quick note on shared databases for human and other commonly used model organisms. The O2 cluster has a designated directory at <code class="language-plaintext highlighter-rouge">/n/groups/shared_databases/</code> in which there are files that can be accessed by any user. These files contain, but are not limited to, genome indices for various tools, reference sequences, tool specific data, and data from public databases, such as NCBI and PDB. So when using a tool that requires a reference of sorts, it is worth taking a quick look here because chances are it’s already been taken care of for you.</p>

  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">ls</span> <span class="nt">-l</span> /n/groups/shared_databases/igenome/
</code></pre></div>  </div>
</blockquote>

<h4 id="creating-a-genome-index">Creating a genome index</h4>

<p>For this workshop we have generated the genome indices for you, so that we don’t get held up waiting on the generation of the indices (it takes a while and requires a lot of memory). The index can be found in <code class="language-plaintext highlighter-rouge">/data/NICHD-core0/references/human/gencode-v28/genome/star/human_gencode-v28</code></p>

<p>The command to create an index can be found in the job submission script we have linked <a href="/modified-intro-to-rnaseq-hpc/modified-intro-march2025/scripts/star_genome_index.run">here</a></p>

<blockquote>
  <p><strong>NOTE:</strong> By default the latest human genome build, GRCh38, contains information about alternative alleles for various locations on the genome. If using this version of the GRCh38 genome then it is advisable to use the HISAT2 aligner as it is able to utilize this information during the alignment. There is a version of GRCh38 available that does not have these alleles represented, which is the appropriate version to use with STAR. This is because STAR does not have the functionality to appropriately deal with the presence of alternate alleles as yet.</p>
</blockquote>

<h4 id="aligning-reads">Aligning reads</h4>

<p>Since we already have the reference index ready, we can move on to aligning reads to the genome.</p>

<p>Create an output directory for our alignment files:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq/raw_data

<span class="nv">$ </span><span class="nb">mkdir</span> ../results/STAR
</code></pre></div></div>

<p>For now, we’re going to work on just one sample to set up our workflow. To start we will use the first replicate in the Mov10 over-expression group, <code class="language-plaintext highlighter-rouge">Mov10_oe_1.subset.fq</code>. Details on STAR and its functionality can be found in the <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">user manual</a>; we encourage you to peruse through to get familiar with all available options.</p>

<p>The basic options for aligning reads to the genome using STAR are:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--runThreadN</code>: number of threads / cores</li>
  <li><code class="language-plaintext highlighter-rouge">--readFilesIn</code>: /path/to/FASTQ_file</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeDir</code>: /path/to/genome_indices_directory</li>
  <li><code class="language-plaintext highlighter-rouge">--outFileNamePrefix</code>: prefix for all output files</li>
</ul>

<p>Listed below are additional parameters that we will use in our command:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--outSAMtype</code>: output filetype (SAM default)</li>
  <li><code class="language-plaintext highlighter-rouge">--outSAMunmapped</code>: what to do with unmapped reads</li>
</ul>

<blockquote>
  <p><strong>NOTE:</strong> Note that “<strong>STAR’s default parameters are optimized for mammalian genomes.</strong> Other species may require significant modifications of some alignment parameters; in particular, the maximum and minimum intron sizes have to be reduced for organisms with smaller introns” [<a href="http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635.full.pdf+html">1</a>].</p>
</blockquote>

<p>The full command is provided below for you to copy paste into your terminal. If you want to manually enter the command, it is advisable to first type out the full command in a text editor (i.e. <a href="http://www.sublimetext.com/">Sublime Text</a> or <a href="https://notepad-plus-plus.org/">Notepad++</a>) on your local machine and then copy paste into the terminal. This will make it easier to catch typos and make appropriate changes.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># assumes you are in raw_data directory</span>
<span class="nv">$ </span>STAR <span class="nt">--genomeDir</span> /data/NICHD-core0/references/human/gencode-v28/genome/star/human_gencode-v28 <span class="se">\</span>
<span class="nt">--runThreadN</span> 6 <span class="se">\</span>
<span class="nt">--readFilesIn</span> Mov10_oe_1.subset.fq <span class="se">\</span>
<span class="nt">--outFileNamePrefix</span> ../results/STAR/Mov10_oe_1_ <span class="se">\</span>
<span class="nt">--outSAMtype</span> BAM SortedByCoordinate <span class="se">\</span>
<span class="nt">--outSAMunmapped</span> Within <span class="se">\</span>
<span class="nt">--outSAMattributes</span> Standard 
</code></pre></div></div>

<p>Since this will take longer with a full-size human reference genome, let’s submit an SBATCH script:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#!/bin/bash</span>

<span class="c">#SBATCH --partition=quick</span>
<span class="c">#SBATCH --time=03:00:00       # time limit</span>
<span class="c">#SBATCH --cpus-per-task=6        # number of cores</span>
<span class="c">#SBATCH --mem=32g   # requested memory</span>
<span class="c">#SBATCH --job-name salmon_in_serial      # Job name</span>
<span class="c">#SBATCH -o %j.out           # File to which standard output will be written</span>
<span class="c">#SBATCH -e %j.err       # File to which standard error will be written</span>
<span class="c">#SBATCH --mail-type=BEGIN,END</span>

<span class="c"># Load Salmon module</span>
module load star/2.7.6a

<span class="c"># Change directory to where the data is</span>
<span class="nb">cd</span> /data/NICHD-core0/test/changes/rc_training/rnaseq/raw_data

STAR <span class="nt">--genomeDir</span> /data/NICHD-core0/references/human/gencode-v28/genome/star/human_gencode-v28 <span class="nt">--runThreadN</span> 6 <span class="nt">--readFilesIn</span> Mov10_oe_1.subset.fq <span class="nt">--outFileNamePrefix</span> ../results/STAR/Mov10_oe_1_ <span class="nt">--outSAMtype</span> BAM SortedByCoordinate <span class="nt">--outSAMunmapped</span> Within <span class="nt">--outSAMattributes</span> Standard 
</code></pre></div></div>

<h4 id="star-output">STAR output</h4>

<p>After running our single FASTQ file through the STAR aligner, you should have a number of output files in the <code class="language-plaintext highlighter-rouge">~/rnaseq/results/STAR</code> directory. Let’s take a quick look at some of the files that were generated and explore their content.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq/results/STAR
    
<span class="nv">$ </span><span class="nb">ls</span> <span class="nt">-lh</span>
</code></pre></div></div>

<p>You should have <strong>5 output files</strong> plus a single <code class="language-plaintext highlighter-rouge">.tmp</code> directory for the Mov10_oe_1 sample. The contents of the 5 files are described below:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">Log.final.out</code> - a summary of mapping statistics for the sample</li>
  <li><code class="language-plaintext highlighter-rouge">Aligned.sortedByCoord.out.bam</code> - the aligned reads, sorted by coordinate, in BAM format</li>
  <li><code class="language-plaintext highlighter-rouge">Log.out</code> - a running log from STAR, with information about the run</li>
  <li><code class="language-plaintext highlighter-rouge">Log.progress.out</code> - job progress with the number of processed reads, % of mapped reads etc., updated every ~1 minute</li>
  <li><code class="language-plaintext highlighter-rouge">SJ.out.tab</code> - high confidence collapsed splice junctions in tab-delimited format. Only junctions supported by uniquely mapping reads are reported</li>
</ul>

<p><strong>We are most interested in the BAM file, which will allow us to proceed with Qualimap.</strong></p>

<h3 id="running-qualimap">Running Qualimap</h3>

<p>Alignment data files frequently contain biases that are introduced by sequencing technologies and/or during sample preparation. Therefore, one of the fundamental requirement during analysis of these data is to perform quality control. In this way, we get an idea of <strong>how well our reads align to the reference</strong> and <strong>how well data fit with the expected outcome</strong>.</p>

<p>We will take the BAM file we generated in the previous step and use it as input to Qualimap which computes various quality metrics such as DNA or rRNA contamination, 5’-3’ biases, and coverage biases.</p>

<p>To <strong>run Qualimap</strong>, change directories to the <code class="language-plaintext highlighter-rouge">rnaseq</code> folder and make a <code class="language-plaintext highlighter-rouge">qualimap</code> folder inside the <code class="language-plaintext highlighter-rouge">results</code> directory:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq

<span class="nv">$ </span><span class="nb">mkdir</span> <span class="nt">-p</span> results/qualimap
</code></pre></div></div>

<p>By default, Qualimap will try to open a GUI to run Qualimap, so we need to run the <code class="language-plaintext highlighter-rouge">unset DISPLAY</code> command:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">unset </span>DISPLAY
</code></pre></div></div>

<p>We also need to load the qualimap module:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>module load qualimap/2.2.1
</code></pre></div></div>

<p>Now we are ready to run Qualimap on our BAM file! There are different tools or modules available through Qualimap, and the <a href="http://qualimap.bioinfo.cipf.es/doc_html/command_line.html">documentation website</a> details the tools and options available. We are interested in the <code class="language-plaintext highlighter-rouge">rnaseq</code> tool. To see the arguments available for this tool we can search the help:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>qualimap rnaseq 
</code></pre></div></div>

<p>We will be running Qualimap with the following specifications:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">-outdir</code>: output directory for html report</li>
  <li><code class="language-plaintext highlighter-rouge">-a</code>: Counting algorithm - uniquely-mapped-reads(default) or proportional (each multi-mapped read is weighted according to the number of mapped locations)</li>
  <li><code class="language-plaintext highlighter-rouge">-bam</code>: path/to/bam/file(s)</li>
  <li><code class="language-plaintext highlighter-rouge">-p</code>: Sequencing library protocol - strand-specific-forward, strand-specific-reverse or non-strand-specific (default)</li>
  <li><code class="language-plaintext highlighter-rouge">-gtf</code>: path/to/gtf/file - <strong>needs to match the genome build and GTF used in alignment</strong></li>
  <li><code class="language-plaintext highlighter-rouge">--java-mem-size=</code>: set Java memory</li>
</ul>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># assuming you are running the script from /results</span>
qualimap rnaseq <span class="nt">-outdir</span> results/qualimap/Mov10_oe_1 <span class="nt">-a</span> proportional /
<span class="nt">-bam</span> STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam /
<span class="nt">-p</span> strand-specific-reverse /
<span class="nt">-gtf</span> /data/NICHD-core0/references/human/gencode-v28/annotation/human_gencode-v28.gtf <span class="nt">--java-mem-size</span><span class="o">=</span>8G
</code></pre></div></div>

<h3 id="the-qualimap-report">The Qualimap report</h3>

<p>The Qualimap report in HTML format should be present in the <code class="language-plaintext highlighter-rouge">results/qualimap</code> directory. To view this report you need a web browser, so you would need to transfer it over to your laptop. You can do so by Connecting to Server again.</p>

<p>However, you don’t need to do that. We generated this report on a subset of data, to get a better idea of the metrics let’s <strong>take a look at the report of the full dataset for <code class="language-plaintext highlighter-rouge">Mov_oe_1</code></strong>, available <a href="https://www.dropbox.com/scl/fi/t3t41hjubytbce6d9g29h/Mov10_oe_1_fulldata_qualimap.zip?rlkey=aum3nw3jym3t4f3l31p6564jn&amp;dl=1">in this zipped folder</a>. Please download and unzip the folder; find the HTML report within and open it in your browser.</p>

<h4 id="read-alignment-summary"><strong>Read alignment summary</strong></h4>

<p>The first few numbers listed in the report are the mapping statistics. Qualimap also computes counts by assigning reads to genes and <a href="http://qualimap.conesalab.org/doc_html/analysis.html#id7">reports associated statistics</a>. For example it computes the following:</p>

<ul>
  <li>
    <p>the number of reads aligned to genes</p>
  </li>
  <li>
    <p>number of ambiguous alignments (reads matching several genes)</p>
  </li>
  <li>
    <p>number of alignments without any feature (intronic and intergenic)</p>
  </li>
</ul>

<p><img src="../img/qualimap_read_alignment.png" width="700" /></p>

<blockquote>
  <ul>
    <li>The percentage of mapped reads is a global indicator of the overall sequencing accuracy. We expect between 70-90% of reads to be mapped for the human genome.</li>
    <li>Expect a small fraction of reads to be mapping equally well to multiple regions in the genome (‘multi-mapping reads’).</li>
    <li>The count related metrics are not as relevant to us since we have quantified with Salmon at the transcript level.</li>
  </ul>
</blockquote>

<h4 id="genomic-origin-of-reads-exon-intron-or-intergenic-region"><strong>Genomic Origin of Reads: Exon, Intron or Intergenic Region?</strong></h4>

<p>This section reports how many alignments fall into exonic, intronic and intergenic regions along with a number of intronic/intergenic alignments overlapping exons. Exonic region includes 5’UTR,protein coding region and 3’UTR region. This information is summarized in table in addition to a pie chart as shown below.</p>

<p><img src="../img/qualimap_genomic_feature.png" width="700" /></p>

<blockquote>
  <ul>
    <li>Even if you have high genomic mapping rate for all samples, check to see where the reads are mapping. Expect a high proportion of reads mapping to exonic regions (&gt; 60%) and lower intronic mapping rates (20 -30%).</li>
    <li>A higher intronic mapping rate is expected for rRNA removal compared to polyA selection. The intronic reads likely originate from immature transcripts which include either full-length pre-mRNA molecules or nascent transcripts where the RNA polymerase has not yet attached to the 3′ end of the gene.</li>
    <li>A roughly equal distribution of reads mapping to intronic, exonic and intergenic regions suggests that there is DNA contamination.</li>
    <li>Ribosomal RNA (rRNA) constitutes a large majority of the RNA species in any total RNA preparation. Despite depletion methods, you can never achieve complete rRNA removal. Even with Poly-A enrichment a small percentage of ribosomal RNA can stick to the enrichment beads non-specifically. Excess ribosomal content (&gt; 2%) will normally have to be filtered out so that differences in rRNA mapped reads across samples do not affect alignment rates and skew subsequent normalization of the data.</li>
  </ul>
</blockquote>

<h4 id="transcript-coverage-profile"><strong>Transcript coverage profile</strong></h4>

<p>The profile provides ratios between mean coverage at the 5’ region, the 3’ region and the whole transcript. Coverage plots are generated for all genes total, and also for the 500 highest-expressed and 500 lowest-expressed genes separately.</p>

<ul>
  <li><strong>5’ bias</strong>: the ratio between mean coverage at the 5’ region (first 100bp) and the whole transcript</li>
  <li><strong>3’ bias</strong>: is the ratio between mean coverage at the 3’ region (last 100bp) and the whole transcript</li>
  <li><strong>5’-3’ bias</strong>: is the ratio between both biases.</li>
</ul>

<p align="center">

<img src="../img/qualimap_transcript_coverage.png" width="700" />

</p>

<p align="center">

<img src="../img/qualimap_coverage_profile.png" width="700" />

</p>

<blockquote>
  <ul>
    <li>In a perfect sequencing experiment you would expect to see a 5’-3’ bias ratio of 1 with low coverage at both ends of the transcript. This would suggest no bias is present.</li>
    <li>It is well-documented that libraries prepared with polyA selection can lead to high expression in 3’ region (3’ bias). At least <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-6-r86">one study</a> shows the reverse effect for rRNA removal.</li>
    <li>If reads primarily accumulate at the 3’ end of transcripts in poly(A)-selected samples, this might indicate low RNA quality in the starting material.</li>
  </ul>
</blockquote>

<h4 id="junction-analysis"><strong>Junction Analysis</strong></h4>

<p>Qualimap also reports the total number of reads mapping to splice junctions and the 10 most frequent junction rates. The pie chart shows analysis of junction positions in spliced alignments.</p>

<ul>
  <li>Known category represents percentage of alignments where both junction sides are known.</li>
  <li>Partly known represents alignments where only one junction side is known.</li>
  <li>All other alignments with junctions are marked as Novel.</li>
</ul>

<p align="center">

<img src="../img/qualimap_junctions.png" width="700" />

</p>

<blockquote>
  <p>Other tools like <a href="https://software.broadinstitute.org/cancer/cga/rna-seqc">RNASeQC</a> will plot figures that can help evaluate <strong>GC content bias</strong>. This is also an important aspect of QC, as low/high GC content regions will tend to have low coverage.</p>
</blockquote>

<hr />

<h2 id="summary">Summary</h2>

<p>Taken together, these metrics give us some insight into the quality of our samples and help us in identifying any biases present in our data. The conclusions derived from these QC results may indicate that we need to correct for these biases and so you may want to go back and modify the parameters for Salmon (mapping) accordingly.</p>

<hr />

<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
