<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Counting reads | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="Counting reads" />
<meta name="author" content="Meeta Mistry, Bob Freeman, Radhika Khetani" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/counting_reads.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/counting_reads.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-06-06T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Counting reads" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Meeta Mistry, Bob Freeman, Radhika Khetani"},"dateModified":"2017-06-06T00:00:00+00:00","datePublished":"2017-06-06T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"Counting reads","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/counting_reads.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/modified-intro-march2025/lessons/counting_reads.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/modified-intro-march2025/assets/css/style.css?v=b507c6c38fbc65ecc2e6abea386dbda0aeab8b3b">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/modified-intro-march2025/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">Counting reads</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <p>Approximate time: 75 minutes</p>

<h2 id="learning-objectives">Learning Objectives:</h2>

<ul>
  <li>understand how counting tools work</li>
  <li>generate a count matrix using featureCounts</li>
</ul>

<h2 id="counting-reads-as-a-measure-of-gene-expression">Counting reads as a measure of gene expression</h2>
<p><img src="../img/counts-workflow.png" width="400" /></p>

<p>Once we have our reads aligned to the genome, the next step is to count how many reads have mapped to each gene. There are many tools that can use BAM files as input and output the number of reads (counts) associated with each feature of interest (genes, exons, transcripts, etc.). 2 commonly used counting tools are <a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a> and <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html">htseq-count</a>.</p>

<ul>
  <li>
    <p>The above tools only report the “raw” counts of reads that <strong>map to a single location</strong> (uniquely mapping) and are best at counting at the <strong>gene level</strong>. Essentially, total read count associated with a gene (<em>meta-feature</em>) = the sum of reads associated with each of the exons (<em>feature</em>) that “belong” to that gene.</p>
  </li>
  <li>
    <p>There are <strong>other tools</strong> available that are able to account for <strong>multiple transcripts</strong> for a given gene. In this case the counts are not whole numbers, but have fractions. In the simplest example case, if 1 read is associated with 2 transcripts, it can get counted as 0.5 and 0.5 and the resulting count for that transcript is not a whole number.</p>
  </li>
  <li>
    <p>In addition there are <strong>other tools that will count multimapping reads</strong>, but this is a dangerous thing to do since you will be overcounting the total number of reads which can cause issues with normalization and eventually with accuracy of differential gene expression results.</p>
  </li>
</ul>

<p><strong>Input for counting = multiple BAM files + 1 GTF file</strong></p>

<p>Simply speaking, the genomic coordinates of where the read is mapped (BAM) are cross-referenced with the genomic coordinates of whichever feature you are interested in counting expression of (GTF), it can be exons, genes or transcripts.</p>

<p><img src="../img/count-fig2.png" width="600" /></p>

<p><strong>Output of counting = A count matrix, with genes as rows and samples are columns</strong></p>

<p>These are the “raw” counts and will be used in statistical programs downstream for differential gene expression.</p>

<p><img src="../img/count-matrix.png" width="300" /></p>

<h3 id="counting-using-featurecounts">Counting using featureCounts</h3>
<p>Today, we will be using the <a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a> tool to get the <em>gene</em> counts. We picked this tool because it is accurate, fast and is relatively easy to use. It counts reads that map to a single location (uniquely mapping) and follows the scheme in the figure below for assigning reads to a gene/exon.</p>

<p><img src="../img/union.png" width="300" /></p>

<p>featureCounts can also take into account whether your data are <strong>stranded</strong> or not. If strandedness is specified, then in addition to considering the genomic coordinates it will also take the strand into account for counting. If your data are stranded always specify it.</p>

<h4 id="setting-up-to-run-featurecounts">Setting up to run featureCounts</h4>
<p>First things first, start an interactive session with 4 cores:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>srun <span class="nt">--pty</span> <span class="nt">-p</span> interactive <span class="nt">-t</span> 0-12:00 <span class="nt">-n</span> 4 <span class="nt">--mem</span> 8G <span class="nt">--reservation</span><span class="o">=</span>HBC1 /bin/bash
</code></pre></div></div>

<p>Now, change directories to your rnaseq directory and start by creating 2 directories, (1) a directory for the output and (2) a directory for the bam files:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq/
<span class="nv">$ </span><span class="nb">mkdir </span>results/counts results/STAR/bams
</code></pre></div></div>

<p>Rather than using the BAM file we generated in the last lesson, let’s copy over all of the BAM files that we have already generated for you:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
<span class="nv">$ </span><span class="nb">cp</span> /n/groups/hbctraining/intro_rnaseq_hpc/bam_STAR/<span class="k">*</span>bam ~/rnaseq/results/STAR/bams
</code></pre></div></div>
<p>featureCounts is not available as a module on O2, but we have already added the path for it to our <code class="language-plaintext highlighter-rouge">$PATH</code> variable last time.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">echo</span> <span class="nv">$PATH</span>  <span class="c"># You should see /n/app/bcbio/tools/bin/ among other paths</span>
</code></pre></div></div>

<blockquote>
  <p>** If you don’t see <code class="language-plaintext highlighter-rouge">/n/app/bcbio/tools/bin/</code> in your <code class="language-plaintext highlighter-rouge">$PATH</code> variable, add the following <code class="language-plaintext highlighter-rouge">export</code> command to your <code class="language-plaintext highlighter-rouge">~/.bashrc</code> file using vim: <code class="language-plaintext highlighter-rouge">export PATH=/n/app/bcbio/tools/bin/:$PATH</code>.**</p>
</blockquote>

<h4 id="running-featurecounts">Running featureCounts</h4>

<p>How do we use this tool, what is the command and what options/parameters are available to us?</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>featureCounts
</code></pre></div></div>

<p>So, it looks like the usage is <code class="language-plaintext highlighter-rouge">featureCounts [options] -a &lt;annotation_file&gt; -o &lt;output_file&gt; input_file1 [input_file2] ... </code>, where <code class="language-plaintext highlighter-rouge">-a</code>, <code class="language-plaintext highlighter-rouge">-o</code> and input files are required.</p>

<p>We are going to use the following options:</p>

<p><code class="language-plaintext highlighter-rouge">-T 4 # specify 4 cores</code></p>

<p><code class="language-plaintext highlighter-rouge">-s 2 # these data are "reverse"ly stranded</code></p>

<p>and the following are the values for the required parameters:</p>

<p><code class="language-plaintext highlighter-rouge">-a ~/rnaseq/reference_data/chr1-hg19_genes.gtf # required option for specifying path to GTF</code></p>

<p><code class="language-plaintext highlighter-rouge">-o ~/rnaseq/results/counts/Mov10_featurecounts.txt # required option for specifying path to, and name of the text output (count matrix)</code></p>

<p><code class="language-plaintext highlighter-rouge">~/rnaseq/results/STAR/bams/*bam # the list of all the bam files we want to collect count information for</code></p>

<p>Let’s run this now:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>featureCounts <span class="nt">-T</span> 4 <span class="nt">-s</span> 2 <span class="se">\</span>
  <span class="nt">-a</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/Homo_sapiens.GRCh38.92.gtf <span class="se">\</span>
  <span class="nt">-o</span> ~/rnaseq/results/counts/Mov10_featurecounts.txt <span class="se">\</span>
  ~/rnaseq/results/STAR/bams/<span class="k">*</span>.out.bam
</code></pre></div></div>

<blockquote>
  <p>If you wanted to collect the information that is on the screen as the job runs, you can modify the command and add the <code class="language-plaintext highlighter-rouge">2&gt;</code> redirection at the end. This type of redirection will collect all the information from the terminal/screen into a file.</p>
</blockquote>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># **DO NOT RUN THIS** </span>
<span class="c"># note the last line of the command below</span>
	
<span class="nv">$ </span>featureCounts <span class="nt">-T</span> 4 <span class="nt">-s</span> 2 <span class="se">\</span>
  <span class="nt">-a</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/Homo_sapiens.GRCh38.92.gtf <span class="se">\</span>
  <span class="nt">-o</span> ~/rnaseq/results/counts/Mov10_featurecounts.txt <span class="se">\</span>
  ~/rnaseq/results/STAR/bams/<span class="k">*</span>.out.bam <span class="se">\</span>
  2&gt; /unix_lesson/rnaseq/results/counts/Mov10_featurecounts.screen-output
</code></pre></div></div>
<h4 id="featurecounts-output">featureCounts output</h4>

<p>The output of this tool is 2 files, <em>a count matrix</em> and <em>a summary file</em> that tabulates how many the reads were “assigned” or counted and the reason they remained “unassigned”. Let’s take a look at the summary file:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>less results/counts/Mov10_featurecounts.txt.summary
</code></pre></div></div>
<p>Now let’s look at the count matrix:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>less results/counts/Mov10_featurecounts.txt
</code></pre></div></div>

<h5 id="cleaning-up-the-featurecounts-matrix">Cleaning up the featureCounts matrix</h5>
<p>There is information about the genomic coordinates and the length of the gene, we don’t need this for the next step, so we are going to extract the columns that we are interested in.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cut</span> <span class="nt">-f1</span>,7,8,9,10,11,12 results/counts/Mov10_featurecounts.txt <span class="o">&gt;</span> results/counts/Mov10_featurecounts.Rmatrix.txt
</code></pre></div></div>
<p>The next step is to clean it up a little further by modifying the header line (we could also do this in R, or in a GUI text editor):</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>vim results/counts/Mov10_featurecounts.Rmatrix.txt
</code></pre></div></div>

<p>Vim has nice shortcuts for cleaning up the header of our file using the following steps:</p>

<ol>
  <li>Move the cursor to the beginning of the document by typing: <code class="language-plaintext highlighter-rouge">gg</code> (in command mode).</li>
  <li>Remove the first line by typing: <code class="language-plaintext highlighter-rouge">dd</code> (in command mode).</li>
  <li>Remove the file name following the sample name by typing: <code class="language-plaintext highlighter-rouge">:%s/_Aligned.sortedByCoord.out.bam//g</code> (in command mode).</li>
  <li>
    <p>Remove the path leading up to the file name by typing: <code class="language-plaintext highlighter-rouge">:%s/\/home\/username\/unix_lesson\/rnaseq\/results\/STAR\/bams\///g</code> (in command mode).</p>

    <blockquote>
      <p>Note that we have a <code class="language-plaintext highlighter-rouge">\</code> preceding each <code class="language-plaintext highlighter-rouge">/</code>, which tells vim that we are not using the <code class="language-plaintext highlighter-rouge">/</code> as part of our search and replace command, but instead the <code class="language-plaintext highlighter-rouge">/</code> is part of the pattern that we are replacing. This is called <em>escaping</em> the <code class="language-plaintext highlighter-rouge">/</code>.</p>
    </blockquote>
  </li>
</ol>

<h3 id="note-on-counting-pe-data">Note on counting PE data</h3>

<p>For paired-end (PE) data, the bam file contains information about whether both read1 and read2 mapped and if they were at roughly the correct distance from each other, that is to say if they were “properly” paired. For most counting tools, <strong>only properly paired reads are considered by default, and each read pair is counted only once as a single “fragment”</strong>.</p>

<p>For counting PE fragments associated with genes, the input bam files need to be sorted by read name (i.e. alignment information about both read pairs in adjoining rows). The alignment tool might sort them for you, but watch out for how the sorting was done. If they are sorted by coordinates (like with STAR), you will need to use <code class="language-plaintext highlighter-rouge">samtools sort</code> to re-sort them by read name before using as input in featureCounts. If you do not sort you BAM file by read name before using as input, featureCounts assumes that almost all the reads are not properly paired.</p>

<hr />
<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
