<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Quality control using FASTQC - script running | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="Quality control using FASTQC - script running" />
<meta name="author" content="Modified by Sally Chang @ NICHD" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/06_qc_running_fastqc_sbatch.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/06_qc_running_fastqc_sbatch.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2024-10-03T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Quality control using FASTQC - script running" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Modified by Sally Chang @ NICHD"},"dateModified":"2024-10-03T00:00:00+00:00","datePublished":"2024-10-03T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"Quality control using FASTQC - script running","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/06_qc_running_fastqc_sbatch.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/06_qc_running_fastqc_sbatch.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/assets/css/style.css?v=d838fbb944b8d2e4dc973fe8850414c54cfcdf99">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">Quality control using FASTQC - script running</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <h2 id="learning-objectives">Learning Objectives:</h2>

<ul>
  <li>Create and run a SLURM job submission script to automate quality assessment</li>
</ul>

<h2 id="quality-control-of-fastq-files">Quality Control of FASTQ files</h2>

<h3 id="performing-quality-assessment-using-job-submission-scripts">Performing quality assessment using job submission scripts</h3>

<p>So far in our FASTQC analysis, we have been directly submitting commands to Biowulf using an interactive session (ie. <code class="language-plaintext highlighter-rouge">fastqc -o ../results/fastqc/ -t 6 *.fq</code>). However, we can submit a command or series of commands to these partitions using job submission scripts. <strong><em>There might be a number of reasons we would like to be able to do this…EXPAND HERE</em></strong></p>

<p><strong>Job submission scripts</strong> for Biowulf are just regular shell scripts, but contain the Slurm <strong>options/directives</strong> for our job submission. These directives define the various resources we are requesting for our job (i.e <em>number of cores, name of partition, runtime limit</em> )</p>

<p>Submission of the script using the <code class="language-plaintext highlighter-rouge">sbatch</code> command allows Slurm to run your job when its your turn. Let’s create a job submission script to automate what we have done in <a href="/modified-intro-to-rnaseq-hpc/lessons/05_qc_running_fastqc_interactively.html">previous lesson</a> &lt;- <strong><em>MODIFY THIS LINK TO WHATEVER OURS WILL BE.</em></strong></p>

<p>Our script will do the following:</p>

<ol>
  <li>Change directories to where the FASTQ files are located</li>
  <li>Load the FastQC module</li>
  <li>Run FastQC on all of our FASTQ files</li>
</ol>

<p>Let’s first change the directory to <code class="language-plaintext highlighter-rouge">/rnaseq/scripts</code>, and create a script named <code class="language-plaintext highlighter-rouge">mov10_fastqc.run</code> using <code class="language-plaintext highlighter-rouge">vim</code>.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd </span>rnaseq/scripts

<span class="nv">$ </span>vim mov10_fastqc.run
</code></pre></div></div>

<p>Once in the vim editor, click <code class="language-plaintext highlighter-rouge">i</code> to enter INSERT mode. The first thing we need in our script is the <strong>shebang line</strong>:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#!/bin/bash</span>
</code></pre></div></div>

<p>Following the shebang line are the Slurm directives. For the script to run, we need to include options for <strong>queue/partition (–partition) and runtime limit (–time)</strong>. To specify our options, we precede the option with <code class="language-plaintext highlighter-rouge">#SBATCH</code>. Some key resources to specify are:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Resource</th>
      <th style="text-align: center">Flag</th>
      <th style="text-align: center">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">partition</td>
      <td style="text-align: center">–partition</td>
      <td style="text-align: center">partition name</td>
    </tr>
    <tr>
      <td style="text-align: center">time</td>
      <td style="text-align: center">–time</td>
      <td style="text-align: center">hours:minutes run limit, after which the job will be killed</td>
    </tr>
    <tr>
      <td style="text-align: center">core</td>
      <td style="text-align: center">-cpus-per-task</td>
      <td style="text-align: center">number of cores requested – this needs to be greater than or equal to the number of cores you plan to use to run your job</td>
    </tr>
    <tr>
      <td style="text-align: center">memory</td>
      <td style="text-align: center">–mem</td>
      <td style="text-align: center">memory limit per compute node for the job</td>
    </tr>
  </tbody>
</table>

<p>Let’s specify those options as follows:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#SBATCH --partition=quick        # Based on running these analyses before, we can get away with running this on a quick (priority!) node</span>
<span class="c">#SBATCH --time=02:00:00       # time limit</span>
<span class="c">#SBATCH --cpus-per-task=6        # number of cores</span>
<span class="c">#SBATCH --mem=6g   # requested memory</span>
<span class="c">#SBATCH --job-name rnaseq_mov10_fastqc      # Job name</span>
<span class="c">#SBATCH -o %j.out           # File to which standard output will be written</span>
<span class="c">#SBATCH -e %j.err       # File to which standard error will be written</span>
</code></pre></div></div>

<p>Now in the body of the script, we can include any commands we want to run. In this case, it will be the following:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">## Change directories to where the fastq files are located. For now I will use absolute paths until we find a good solution for a prefix. </span>
<span class="nb">cd</span> /data/NICHD-core0/test/changes/rc_training/rnaseq/raw_data

<span class="c">## Load modules required for script commands</span>
module load fastqc/0.12.1

<span class="c">## Run FASTQC</span>
fastqc <span class="nt">-o</span> /data/NICHD-core0/test/changes/rc_training/rnaseq/fastqc/ <span class="nt">-t</span> 6 <span class="k">*</span>.fq
</code></pre></div></div>

<blockquote>
  <p><strong>NOTE:</strong> These are the same commands we used when running FASTQC in the interactive session. Since we are writing them in a script, the <code class="language-plaintext highlighter-rouge">tab</code> completion function will <strong>not work</strong>, so please make sure you don’t have any typos when writing the script!</p>
</blockquote>

<p>Once done with your script, click <code class="language-plaintext highlighter-rouge">esc</code> to exit the INSERT mode. Then save and quit the script by typing <code class="language-plaintext highlighter-rouge">:wq</code>. You may double check your script by typing <code class="language-plaintext highlighter-rouge">less mov10_fastqc.run</code>. If everything looks good submit the job!</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>sbatch mov10_fastqc.run
</code></pre></div></div>

<p>You should immediately see a prompt saying <code class="language-plaintext highlighter-rouge">Submitted batch job JobID</code>. Your job is assigned with that unique identifier <code class="language-plaintext highlighter-rouge">JobID</code>. You can check on the status of your job with:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>squeue <span class="nt">-u</span> changes
</code></pre></div></div>

<p>Look for the row that corresponds to your <code class="language-plaintext highlighter-rouge">JobID</code>. The third column indicates the state of your job. Possible states include <code class="language-plaintext highlighter-rouge">PENDING (PD)</code>, <code class="language-plaintext highlighter-rouge">RUNNING (R)</code>, <code class="language-plaintext highlighter-rouge">COMPLETING (CG)</code>. For this example, once your job is <code class="language-plaintext highlighter-rouge">RUNNING</code>, you should expect it to finish in less than two minutes (although of course this will not always be the case for longer analyses).</p>

<blockquote>
  <p><strong>NOTE:</strong> Here is a <a href="https://hpc.nih.gov/docs/userguide.html#states">table of Job State Codes</a> from the Biowulf User Guide.</p>
</blockquote>

<p>Check out the output files in your directory:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">ls</span> <span class="nt">-lh</span> ../results/fastqc/
</code></pre></div></div>

<p>There should also be one standard error (<code class="language-plaintext highlighter-rouge">.err</code>) and one standard out (<code class="language-plaintext highlighter-rouge">.out</code>) files from the job listed in <code class="language-plaintext highlighter-rouge">~/rnaseq/scripts</code>. You can move these over to your <code class="language-plaintext highlighter-rouge">logs</code> directory and give them more intuitive names:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mv</span> <span class="k">*</span>.err ../logs/fastqc.err
<span class="nv">$ </span><span class="nb">mv</span> <span class="k">*</span>.out ../logs/fastqc.out
</code></pre></div></div>

<blockquote>
  <p><strong>NOTE:</strong> The <code class="language-plaintext highlighter-rouge">.err</code> and <code class="language-plaintext highlighter-rouge">.out</code> files store log information during the script running. They are helpful resources, especially when your script does not run as expected and you need to troubleshoot the script.</p>
</blockquote>

<hr />

<p><strong>Exercise</strong> 1. Take a look at what’s inside the <code class="language-plaintext highlighter-rouge">.err</code> and <code class="language-plaintext highlighter-rouge">.out</code> files. What do you observe? Do you remember where you see those information when using the interactive session? 2. How would you change the <code class="language-plaintext highlighter-rouge">mov10_fastqc.run</code> script if you had 9 fastq files you wanted to run in parallel?</p>

<hr />

<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>

<ul>
  <li><em>The materials used in this lesson was derived from work that is Copyright © Data Carpentry (<a href="http://datacarpentry.org/">http://datacarpentry.org/</a>). All Data Carpentry instructional material is made available under the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0).</em></li>
</ul>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
