<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>RNA-Seq workflow | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="RNA-Seq workflow" />
<meta name="author" content="Mary Piper, Meeta Mistry, Radhika Khetani,  Bob Freeman" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/rnaseq_workflow.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/rnaseq_workflow.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-09-20T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="RNA-Seq workflow" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Mary Piper, Meeta Mistry, Radhika Khetani,  Bob Freeman"},"dateModified":"2017-09-20T00:00:00+00:00","datePublished":"2017-09-20T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"RNA-Seq workflow","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/rnaseq_workflow.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/rnaseq_workflow.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/assets/css/style.css?v=b507c6c38fbc65ecc2e6abea386dbda0aeab8b3b">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">RNA-Seq workflow</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <p>Approximate time: 90 minutes</p>

<h2 id="learning-objectives">Learning Objectives:</h2>

<ul>
  <li>Describe and implement the RNA-seq workflow to align reads to the reference genome</li>
  <li>Describe tools and methods within the RNA-seq workflow</li>
  <li>Assessing input and output filetypes</li>
</ul>

<h2 id="setting-up-to-run-the-rna-seq-workflow">Setting up to run the RNA-seq workflow</h2>

<p>To get started with this lesson, we will start an interactive session and ask for 6 cores, by adding <code class="language-plaintext highlighter-rouge">-n 6</code> to the <code class="language-plaintext highlighter-rouge">srun</code> command:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>srun <span class="nt">--pty</span> <span class="nt">-p</span> interactive <span class="nt">-t</span> 0-12:00 <span class="nt">-n</span> 6 <span class="nt">--mem</span> 8G <span class="nt">--reservation</span><span class="o">=</span>HSPH bash	
</code></pre></div></div>

<p>Change directories into the <code class="language-plaintext highlighter-rouge">unix_lesson</code> directory and copy the <code class="language-plaintext highlighter-rouge">reference_data</code> folder into your project directory:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq
</code></pre></div></div>

<p>You should have a directory tree setup similar to that shown below. It is best practice to have all files you intend on using for your workflow present within the same directory.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>rnaseq/
	├── raw_data/
	├── reference_data/
	   └── chr1.fa
	   └── chr1-hg19_genes.gtf
	├── meta/
	├── results/
	├── scripts/
	└── logs/
</code></pre></div></div>

<p>Below is a general overview of the steps involved in RNA-seq analysis.</p>

<p><img src="../img/RNAseqWorkflow.png" width="400" /></p>

<h2 id="read-alignment">Read Alignment</h2>
<p>The alignment process consists of choosing an appropriate reference genome to map our reads against, and performing the read alignment using one of several splice-aware alignment tools such as <a href="https://github.com/alexdobin/STAR">STAR</a> or <a href="https://ccb.jhu.edu/software/hisat2/index.shtml">HISAT2</a> (HISAT2 is a successor to both HISAT and TopHat2). The choice of aligner is a personal preference and also dependent on the computational resources that are available to you. The splice-aware feature is especially important for RNA-seq data, because we want an alignment tool that considers splice junctions when performing alignments.</p>

<p>For this workshop we will be using STAR (Spliced Transcripts Alignment to a Reference), an aligner designed to specifically address many of the challenges of RNAseq read mapping. STAR is shown to have <strong>high accuracy</strong> and outperforms other aligners by more than a <strong>factor of 50 in mapping speed (but also requires quite a bit of memory</strong>).</p>

<h3 id="star-alignment-strategy">STAR Alignment Strategy</h3>

<p>The algorithm achieves this highly efficient mapping by performing a two-step process:</p>

<ol>
  <li>Seed searching</li>
  <li>Clustering, stitching, and scoring</li>
</ol>

<h4 id="seed-searching">Seed searching</h4>

<p>For every read that STAR aligns, STAR will search for the longest sequence that exactly matches the reference genome:</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step1.png" alt="STAR_step1" /></p>

<p>The different parts of the read that are mapped separately are called ‘seeds’. So the first MMP that is mapped to the genome is called <em>seed1</em>.</p>

<p>STAR will then search again for only the unmapped portion of the read to find the next longest sequence that exactly matches the reference genome, which will be <em>seed2</em>.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step2.png" alt="STAR_step2" /></p>

<p><strong>If STAR does not find an exact matching sequence</strong> for each part of the read due to mismatches or indels, the seed will be extended.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step3.png" alt="STAR_step3" /></p>

<p><strong>If extension does not give a good alignment</strong>, then the poor quality or adapter sequence (or other contaminating sequence) will be soft clipped.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step4.png" alt="STAR_step4" /></p>

<p>This sequential searching of only the unmapped portions of reads underlies the efficiency of the STAR algorithm. STAR uses an uncompressed suffix array (SA) to efficiently search for the longest matching portions of the read, this allows for quick searching against even the largest reference genomes. Other slower aligners use algorithms that often search for the entire read sequence before splitting reads and performing iterative rounds of mapping. More details on the algorithm itself can be found in the <a href="http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635">STAR publication</a>.</p>

<h4 id="clustering-stitching-and-scoring">Clustering, stitching, and scoring</h4>

<p>The separate seeds are stitched together to create a complete read by first clustering the seeds together based on proximity to a set of seeds that have good alignment scores and are not multi-mapping.</p>

<p>Then the seeds are stitched together based on the best alignment for the read (scoring based on mismatches, indels, gaps, etc.).</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step5.png" alt="STAR_step5" /></p>

<h3 id="setting-up">Setting up</h3>

<p>Let’s get started by loading up the STAR module required for alignment:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>module load star/2.5.2b
</code></pre></div></div>
<p>Create an output directory for our alignment files:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir </span>results/STAR
</code></pre></div></div>
<p>In the automation script, we will eventually loop over all of our files and have the cluster work on the files in parallel. For now, we’re going to work on just one to test and set up our workflow. To start we will use the first replicate in the Mov10 overexpression group, <code class="language-plaintext highlighter-rouge">Mov10_oe_1_subset.fq</code>.</p>

<h3 id="running-star">Running STAR</h3>

<p>Aligning reads using STAR is a two-step process:</p>

<ol>
  <li>Create a genome index</li>
  <li>Map reads to the genome</li>
</ol>

<h4 id="creating-a-genome-index">Creating a genome index</h4>

<p>The genome index is analagous to an index in a textbook, it is used to organize the genome in a way that can be easily accessed and makes for more efficient searching. For this workshop, we have already indexed the reference genome for you as this can take a while. We have provided the code below that you would use to index the genome for your future reference, but please <strong>do not run the code below</strong>. For indexing the reference genome, a reference genome (FASTA) is required and an annotation file (GTF or GFF3) is suggested a more accurate alignment of the reads.</p>

<p>The basic options to <strong>generate genome indices</strong> using STAR are as follows:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--runThreadN</code>: number of threads</li>
  <li><code class="language-plaintext highlighter-rouge">--runMode</code>: genomeGenerate mode</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeDir</code>: /path/to/store/genome_indices</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeFastaFiles</code>: /path/to/FASTA_file (reference genome)</li>
  <li><code class="language-plaintext highlighter-rouge">--sjdbGTFfile</code>: /path/to/GTF_file (gene annotation)</li>
  <li><code class="language-plaintext highlighter-rouge">--sjdbOverhang</code>: readlength -1</li>
</ul>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="k">**</span> DO NOT RUN<span class="k">**</span>
STAR <span class="nt">--runThreadN</span> 6 <span class="nt">--runMode</span> genomeGenerate <span class="nt">--genomeDir</span> ./ <span class="nt">--genomeFastaFiles</span> chr1.fa <span class="nt">--sjdbGTFfile</span> chr1-hg19_genes.gtf <span class="nt">--sjdbOverhang</span> 99
</code></pre></div></div>
<blockquote>
  <p><strong>NOTE:</strong> For human and other commonly used model organisms, the O2 cluster has a designated directory at <code class="language-plaintext highlighter-rouge">/n/groups/shared_databases/</code> in which there are files that can be accessed by any user. These files contain, but are not limited to, genome indices for various tools, reference sequences, tool specific data, and data from public databases, such as NCBI and PDB. So when using a tool and requires a reference of sorts, it is worth taking a quick look here because chances are it’s already been taken care of for you.</p>
</blockquote>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">ls</span> <span class="nt">-l</span> /n/groups/shared_databases/igenome
</code></pre></div></div>

<p>For this workshop we are using reads that originate from a small subsection of chromosome 1 (~300,000 reads) and so we are using only chr1 as the reference genome. Therefore, we cannot use any of the ready-made indices available in the shared folder. The index we have created is located at <code class="language-plaintext highlighter-rouge">/n/groups/hbctraining/intro_rnaseq_hpc/reference_STAR</code> if you wanted to take a look at what files comprise a STAR index.</p>

<h4 id="mapping-reads">Mapping reads</h4>

<p>The basic options for <strong>mapping reads</strong> to the genome using STAR are as follows:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--runThreadN</code>: number of threads</li>
  <li><code class="language-plaintext highlighter-rouge">--readFilesIn</code>: /path/to/FASTQ_file</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeDir</code>: /path/to/genome_indices</li>
  <li><code class="language-plaintext highlighter-rouge">--outFileNamePrefix</code>: prefix for all output files</li>
</ul>

<p>We will also be using some advanced options:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--outSAMtype BAM SortedByCoordinate</code>: output filetype (SAM default)</li>
  <li><code class="language-plaintext highlighter-rouge">--outSAMUnmapped Within</code>: what to do with unmapped reads</li>
</ul>

<p>Note that default filtering is applied in which the maximum number of multiple alignments allowed for a read is set to 10. If a read exceeds this number there is no alignment output. To change the default you can use <code class="language-plaintext highlighter-rouge">--outFilterMultimapNmax</code>, but for this lesson we will leave it as default. The advanced parameters that we are going to use are described below:</p>

<p>More details on STAR and its functionality can be found in the <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">user manual</a>, we encourage you to peruse through to get familiar with all available options.</p>

<p>Now let’s put it all together! The full STAR alignment command is provided below.</p>

<blockquote>
  <p>If you like you can copy-paste it directly into your terminal. Alternatively, you can manually enter the command, but it is advisable to first type out the full command in a text editor (i.e. <a href="http://www.sublimetext.com/">Sublime Text</a> or <a href="https://notepad-plus-plus.org/">Notepad++</a>) on your local machine and then copy paste into the terminal. This will make it easier to catch typos and make appropriate changes.</p>
</blockquote>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>STAR <span class="nt">--runThreadN</span> 6 <span class="se">\</span>
<span class="nt">--genomeDir</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_STAR <span class="se">\</span>
<span class="nt">--readFilesIn</span> raw_data/Mov10_oe_1.subset.fq <span class="se">\</span>
<span class="nt">--outFileNamePrefix</span> results/STAR/Mov10_oe_1_ <span class="se">\</span>
<span class="nt">--outSAMtype</span> BAM SortedByCoordinate <span class="se">\</span>
<span class="nt">--outSAMunmapped</span> Within <span class="se">\</span>
</code></pre></div></div>

<hr />

<p><strong>Exercise</strong></p>

<ul>
  <li>How many files do you see in your output directory?</li>
  <li>Using the <code class="language-plaintext highlighter-rouge">less</code> command take a look at <code class="language-plaintext highlighter-rouge">Mov10_oe_1_Log.final.out</code> and answer the following questions:
    <ol>
      <li>How many reads are uniquely mapped?</li>
      <li>How many reads map to more than 10 locations on the genome?</li>
      <li>How many reads are unmapped due to read length?</li>
    </ol>
  </li>
</ul>

<hr />

<h3 id="alignment-outputs-sambam">Alignment Outputs (SAM/BAM)</h3>

<p>The output we requested from STAR is a BAM file, and by default returns a file in SAM format. <strong>BAM is a binary version of the SAM file, also known as Sequence Alignment Map format.</strong> The SAM file is a tab-delimited text file that contains information for each individual read and its alignment to the genome. The file begins with an optional header (which starts with ‘@’), followed by an alignment section in which each line corresponds to alignment information for a single read. <strong>Each alignment line has 11 mandatory fields</strong> for essential mapping information and a variable number of fields for aligner specific information.</p>

<p>These fields are described briefly below, but for more detailed information the paper by <a href="http://bioinformatics.oxfordjournals.org/content/25/16/2078.full">Heng Li et al</a> is a good start.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/sam_bam.png" alt="SAM" /></p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/sam_bam3.png" alt="SAM" /></p>

<p>Let’s take a quick look at our alignment. To do so we first convert our BAM file into SAM format using samtools and then pipe it to the <code class="language-plaintext highlighter-rouge">less</code> command. This allows us to look at the contents without having to write it to file (since we don’t need a SAM file for downstream analyses).</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>module load gcc/6.2.0 samtools/1.3.1

<span class="nv">$ </span>samtools view <span class="nt">-h</span> results/STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam | less
</code></pre></div></div>

<blockquote>
  <p><strong>NOTE:</strong> When loading the <code class="language-plaintext highlighter-rouge">samtools</code> module we also needed to load the <code class="language-plaintext highlighter-rouge">gcc</code> module. This is because on O2 there are some tools that need the compiler loaded before it can run. When in doubt try <code class="language-plaintext highlighter-rouge">module spider</code> on the tool you are interested in using.</p>
</blockquote>

<p>Scroll through the SAM file and see how the fields correspond to what we expected.</p>

<h3 id="counting-reads">Counting reads</h3>

<p>Once we have our reads aligned to the genome, the next step is to count how many reads have been mapped to each gene. The input files required for counting include the BAM file and an associated gene annotation file in GTF format. <a href="http://htseq.readthedocs.io/en/release_0.9.1/count.html">htseq-count</a> and <a href="http://bioinf.wehi.edu.au/featureCounts/">featureCounts</a> are two commonly used counting tools. Today, we will be using featureCounts to get the <em>gene</em> counts. We picked this tool because it is accurate, fast and is relatively easy to use.</p>

<p><code class="language-plaintext highlighter-rouge">featureCounts</code> works by <strong>taking the alignment coordinates for each read and cross-referencing that to the coordinates for <em>features</em> described in the GTF</strong>. Most commonly a feature is considered to be a gene, which is the union of all exons (which is also a feature type) that make up that gene. <em>Please note that this tool is best used for counting reads associated with <strong>genes</strong>, and not for splice isoforms or transcripts, we will be covering that later today.</em></p>

<p><code class="language-plaintext highlighter-rouge">featureCounts</code> only includes and counts those reads that map to a single location (uniquely mapping) and follows the scheme in the figure below for assigning reads to a gene/exon.</p>

<p><img src="../img/union.png" width="400" /></p>

<p><code class="language-plaintext highlighter-rouge">featureCounts</code> can also take into account whether your data are <strong>stranded</strong> or not. If strandedness is specified, then in addition to considering the genomic coordinates it will also take the strand into account for counting. If your data are stranded always specify it.</p>

<h4 id="setting-up-to-run-featurecounts">Setting up to run featureCounts</h4>

<p>Let’s start by creating a directory for the output:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir </span>results/counts
</code></pre></div></div>
<p><code class="language-plaintext highlighter-rouge">featureCounts</code> is not available as a module on O2, but we have already added the path for it to our <code class="language-plaintext highlighter-rouge">$PATH</code> variable last time.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>which featureCounts  <span class="c"># should return /n/app/bcbio/tools/bin/featureCounts </span>
</code></pre></div></div>

<blockquote>
  <p>** If running the above command does not return <code class="language-plaintext highlighter-rouge">/n/app/bcbio/tools/bin/featureCounts</code>, run <code class="language-plaintext highlighter-rouge">export PATH=/n/app/bcbio/tools/bin:$PATH</code> and try the <code class="language-plaintext highlighter-rouge">which</code> command again.**</p>
</blockquote>

<p>How do we use this tool, what is the command and what options/parameters are available to us?</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>featureCounts
</code></pre></div></div>

<p>So, it looks like the usage is <code class="language-plaintext highlighter-rouge">featureCounts [options] -a &lt;annotation_file&gt; -o &lt;output_file&gt; input_file1 [input_file2] ... </code>, where <code class="language-plaintext highlighter-rouge">-a</code>, <code class="language-plaintext highlighter-rouge">-o</code> and input files are required.</p>

<p>It can also take multiple bam files as input. Since we have only run STAR on 1 FASTQ file, let’s copy over the other bam files that we would need so we can generate the full count matrix.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cp</span> /n/groups/hbctraining/intro_rnaseq_hpc/bam_STAR/<span class="k">*</span>bam ~/rnaseq/results/STAR/
</code></pre></div></div>

<p>We are going to use the following options:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">-T 4 # specify 4 cores</code></li>
  <li><code class="language-plaintext highlighter-rouge">-s 2 # these data are "reverse"ly stranded</code></li>
</ul>

<p>and the following are the values for the required parameters:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">-a ~/rnaseq/reference_data/chr1-hg19_genes.gtf # required option for specifying path to GTF</code></li>
  <li><code class="language-plaintext highlighter-rouge">-o ~/rnaseq/results/counts/Mov10_featurecounts.txt # required option for specifying path to, and name of the text output (count matrix)</code></li>
  <li><code class="language-plaintext highlighter-rouge">~/rnaseq/results/STAR/*bam # the list of all the bam files we want to collect count information for</code></li>
</ul>

<h4 id="running-featurecounts">Running featureCounts</h4>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>featureCounts <span class="nt">-T</span> 4 <span class="nt">-s</span> 2 <span class="se">\</span>
  <span class="nt">-a</span> ~/rnaseq/reference_data/chr1-hg19_genes.gtf <span class="se">\</span>
  <span class="nt">-o</span> ~/rnaseq/results/counts/Mov10_featurecounts.txt <span class="se">\</span>
  ~/rnaseq/results/STAR/<span class="k">*</span>bam
</code></pre></div></div>
<h4 id="featurecounts-output">featureCounts output</h4>

<p>The output of this tool is 2 files, <em>a count matrix</em> and <em>a summary file</em> that tabulates how many the reads were “assigned” or counted and the reason they remained “unassigned”. Let’s take a look at the summary file:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>less results/counts/Mov10_featurecounts.txt.summary
</code></pre></div></div>
<p>Now let’s look at the count matrix:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>less results/counts/Mov10_featurecounts.txt
</code></pre></div></div>
<p>The count matrix that we need to perform differential gene expression analysis needs to look something like this:</p>

<p><img src="../img/count_matrix.png" width="500" /></p>

<p>Since the featureCounts output has additional columns with information about genomic coordinates, gene length etc., we can use the <code class="language-plaintext highlighter-rouge">cut</code> command to select only those columns that you are interested in.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cut</span> <span class="nt">-f1</span>,7,8,9,10,11,12 results/counts/Mov10_featurecounts.txt <span class="o">&gt;</span> results/counts/Mov10_featurecounts.Rmatrix.txt
</code></pre></div></div>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>less results/counts/Mov10_featurecounts.Rmatrix.txt
</code></pre></div></div>

<p>To ready this text file (count matrix) for the next step of differential gene expression analysis, you will need to clean it up further by removing the first header line, and modifying the column names (headers) to simpler, smaller sampleIDs. We can do this using a GUI text editor on our local laptops, or we can try using some of the shortcuts available in <code class="language-plaintext highlighter-rouge">vim</code>!</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>vim results/counts/Mov10_featurecounts.Rmatrix.txt
</code></pre></div></div>

<p>Vim has nice shortcuts for cleaning up the header of our file using the following steps:</p>

<ol>
  <li>Move the cursor to the beginning of the document by typing: <code class="language-plaintext highlighter-rouge">gg</code> (in command mode).</li>
  <li>Remove the first line by typing: <code class="language-plaintext highlighter-rouge">dd</code> (in command mode).</li>
  <li>Remove the file name following the sample name by typing: <code class="language-plaintext highlighter-rouge">:%s/_Aligned.sortedByCoord.out.bam//g</code> (in command mode).</li>
  <li>
    <p>Remove the path leading up to the file name by typing: <code class="language-plaintext highlighter-rouge">:%s/\/home\/rc_training10\/unix_lesson\/rnaseq\/results\/STAR\///g</code> (in command mode).</p>

    <blockquote>
      <p>Note that we have a <code class="language-plaintext highlighter-rouge">\</code> preceding each <code class="language-plaintext highlighter-rouge">/</code>, which tells vim that we are not using the <code class="language-plaintext highlighter-rouge">/</code> as part of our search and replace command, but instead the <code class="language-plaintext highlighter-rouge">/</code> is part of the pattern that we are replacing. This is called <em>escaping</em> the <code class="language-plaintext highlighter-rouge">/</code>.</p>
    </blockquote>
  </li>
</ol>

<blockquote>
  <p><strong>NOTE:</strong> The home directory will need to be changed to either your eCommons ID or the training account you are using.</p>
</blockquote>

<hr />

<blockquote>
  <h4 id="note-on-counting-pe-data">Note on counting PE data</h4>

  <p>For paired-end (PE) data, the bam file contains information about whether both read1 and read2 mapped and if they were at roughly the correct distance from each other, that is to say if they were “properly” paired. For most counting tools, only properly paired reads are considered by default, and each read pair is counted only once as a single “fragment”.</p>

  <p>For counting PE fragments associated with genes, the input bam files need to be sorted by read name (i.e. alignment information about both read pairs in adjoining rows). The alignment tool might sort them for you, but watch out for how the sorting was done. If they are sorted by coordinates (like with STAR), you will need to use <code class="language-plaintext highlighter-rouge">samtools sort</code> to re-sort them by read name before using as input in featureCounts. If you do not sort you BAM file by read name before using as input, featureCounts assumes that almost all the reads are not properly paired.</p>
</blockquote>

<hr />

<h3 id="next-steps-performing-de-analysis-on-the-count-matrix">Next Steps: Performing DE analysis on the count matrix</h3>

<p>This text file with a matrix of raw counts can be used as input to tools like <a href="http://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>, <a href="http://bioconductor.org/packages/release/bioc/html/edgeR.html">EdgeR</a> and <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">limma-voom</a>. Running the DE analysis tools are outside the scope of this workshop since they require a working knowledge of R. We do have <a href="https://github.com/hbctraining/Intro-to-rnaseq-hpc/blob/master/lessons/DE_analysis.md">additional materials</a> that you can go through for performing these analyses using the MOV10 dataset, and we might walk through them if time permits today.</p>

<h3 id="visual-assessment-of-the-alignment">Visual assessment of the alignment</h3>

<p>Index the BAM file for visualization with IGV:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools index results/STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam
</code></pre></div></div>

<p>Use <em><strong>FileZilla</strong></em> to copy the following files to your local machine:</p>

<p><code class="language-plaintext highlighter-rouge">~/unix_lesson/results/STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam</code></p>

<p><code class="language-plaintext highlighter-rouge">~/unix_lesson/results/STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam.bai</code></p>

<blockquote>
  <p><strong>NOTE: You can also transfer files to your laptop using the command line</strong></p>

  <p>Similar to the <code class="language-plaintext highlighter-rouge">cp</code> command, there is a command that allows you to securely copy files between computers. The command is called <code class="language-plaintext highlighter-rouge">scp</code> and allows files to be copied to, from, or between different hosts. It uses ssh for data transfer and provides the same authentication and same level of security as <code class="language-plaintext highlighter-rouge">ssh</code>.</p>

  <p>First, identify the location of the <em>origin file</em> you intend to copy, followed by the <em>destination</em> of that file. Since the original file is located on O2, this requires you to provide remote host and login information.</p>
</blockquote>

<blockquote>
  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>scp user_name@transfer.rc.hms.harvard.edu:/home/user_name/unix_lesson/rnaseq/results/STAR/Mov10_oe_1_Aligned.sortedByCoord.out.bam<span class="k">*</span> /path/to/directory_on_laptop
</code></pre></div>  </div>
</blockquote>

<p><strong>Visualize</strong></p>

<ul>
  <li>Start <a href="https://www.broadinstitute.org/software/igv/download">IGV</a>, <em>you should have this previously installed on your laptop</em>.</li>
  <li>Load the Human genome (hg19) into IGV using the dropdown menu at the top left of your screen. 
<strong>Note</strong>: there is also an option to “Load Genomes from File…” under the “Genomes” pull-down menu - this is useful when working with non-model organisms</li>
  <li>Load the .bam file using the <strong>“Load from File…“</strong> option under the <strong>“File”</strong> pull-down menu. <em>IGV requires the <code class="language-plaintext highlighter-rouge">.bai</code> file to be in the same location as corresponding <code class="language-plaintext highlighter-rouge">.bam</code> file that you want to load into IGV, but there is no other direct use for this index file.</em></li>
</ul>

<p><img src="/modified-intro-to-rnaseq-hpc/img/igv_screenshot.png" alt="IGV screenshot" /></p>

<hr />
<p><strong>Exercise</strong></p>

<p>Now that we have done this for one sample, let’s try using the same commands to perform alignment on one of the control samples. Create an index for the <code class="language-plaintext highlighter-rouge">Irrel_kd_1_Aligned.sortedByCoord.out.bam</code> file and them to your laptop and load into IGV for visualization.</p>

<ol>
  <li>How does the MOV10 gene look in the control sample in comparison to the overexpression sample?</li>
  <li>Take a look at a few other genes by typing into the search bar. For example, PPM1J and PTPN22. How do these genes compare?</li>
</ol>

<hr />

<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>

<ul>
  <li><em>The materials used in this lesson were derived from work that is Copyright © Data Carpentry (http://datacarpentry.org/). 
All Data Carpentry instructional material is made available under the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0).</em></li>
</ul>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
