<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Alignment with STAR | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="Alignment with STAR" />
<meta name="author" content="Meeta Mistry, Bob Freeman, Mary Piper" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/STAR_alignment.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/STAR_alignment.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-06-07T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Alignment with STAR" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Meeta Mistry, Bob Freeman, Mary Piper"},"dateModified":"2017-06-07T00:00:00+00:00","datePublished":"2017-06-07T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"Alignment with STAR","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/STAR_alignment.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/STAR_alignment.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/assets/css/style.css?v=57e2d19053fb95686fdf7b0c6d67a5efc1620535">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">Alignment with STAR</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <p>Approximate time: 90 minutes</p>

<h2 id="learning-objectives">Learning Objectives:</h2>

<ul>
  <li>Understanding the alignment method STAR utilizes to align sequence reads to the reference genome</li>
  <li>Identifying the intricacies of alignment tools used in NGS analysis (parameters, usage, etc)</li>
  <li>Choosing appropriate STAR alignment parameters for our dataset</li>
</ul>

<h2 id="read-alignment">Read Alignment</h2>

<p><img src="../img/RNAseqWorkflow.png" width="400" /></p>

<p>Now that we have explored the quality of our raw reads, we can move on to read alignment. We perform read alignment or mapping to determine where in the genome the reads originated from. The alignment process consists of choosing an appropriate reference genome to map our reads against and performing the read alignment using one of several splice-aware alignment tools such as <a href="http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635">STAR</a> or <a href="http://ccb.jhu.edu/software/hisat2/index.shtml">HISAT2</a>. The choice of aligner is often a personal preference and also dependent on the computational resources that are available to you.</p>

<h2 id="star-aligner">STAR Aligner</h2>

<p>To determine where on the human genome our reads originated from, we will align our reads to the reference genome using <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/">STAR</a> (Spliced Transcripts Alignment to a Reference). STAR is an aligner designed to specifically address many of the challenges of RNA-seq data mapping using a strategy to account for spliced alignments.</p>

<h3 id="star-alignment-strategy">STAR Alignment Strategy</h3>

<p>STAR is shown to have high accuracy and outperforms other aligners by more than a factor of 50 in mapping speed, but it is memory intensive. The algorithm achieves this highly efficient mapping by performing a two-step process:</p>

<ol>
  <li>Seed searching</li>
  <li>Clustering, stitching, and scoring</li>
</ol>

<h4 id="seed-searching">Seed searching</h4>

<p>For every read that STAR aligns, STAR will search for the longest sequence that exactly matches one or more locations on the reference genome. These longest matching sequences are called the Maximal Mappable Prefixes (MMPs):</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step1.png" alt="STAR_step1" /></p>

<p>The different parts of the read that are mapped separately are called ‘seeds’. So the first MMP that is mapped to the genome is called <em>seed1</em>.</p>

<p>STAR will then search again for only the unmapped portion of the read to find the next longest sequence that exactly matches the reference genome, or the next MMP, which will be <em>seed2</em>.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step2.png" alt="STAR_step2" /></p>

<p>This sequential searching of only the unmapped portions of reads underlies the efficiency of the STAR algorithm. STAR uses an uncompressed suffix array (SA) to efficiently search for the MMPs, this allows for quick searching against even the largest reference genomes. Other slower aligners use algorithms that often search for the entire read sequence before splitting reads and performing iterative rounds of mapping.</p>

<p><strong>If STAR does not find an exact matching sequence</strong> for each part of the read due to mismatches or indels, the previous MMPs will be extended.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step3.png" alt="STAR_step3" /></p>

<p><strong>If extension does not give a good alignment</strong>, then the poor quality or adapter sequence (or other contaminating sequence) will be soft clipped.</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step4.png" alt="STAR_step4" /></p>

<h4 id="clustering-stitching-and-scoring">Clustering, stitching, and scoring</h4>

<p>The separate seeds are stitched together to create a complete read by first clustering the seeds together based on proximity to a set of ‘anchor’ seeds, or seeds that are not multi-mapping.</p>

<p>Then the seeds are stitched together based on the best alignment for the read (scoring based on mismatches, indels, gaps, etc.).</p>

<p><img src="/modified-intro-to-rnaseq-hpc/img/alignment_STAR_step5.png" alt="STAR_step5" /></p>

<h2 id="running-star">Running STAR</h2>

<h3 id="set-up">Set-up</h3>

<p>To get started with this lesson, start an interactive session with 6 cores:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>srun <span class="nt">--pty</span> <span class="nt">-p</span> interactive <span class="nt">-t</span> 0-12:00 <span class="nt">-n</span> 6 <span class="nt">--mem</span> 8G <span class="nt">--reservation</span><span class="o">=</span>HBC1 /bin/bash	
</code></pre></div></div>

<p>You should have a directory tree setup similar to that shown below. it is best practice to have all files you intend on using for your workflow present within the same directory. In our case, we have our original FASTQ files generated in the previous section.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>rnaseq
	├── logs
	├── meta
	├── raw_data
	│   ├── Irrel_kd_1.subset.fq
	│   ├── Irrel_kd_2.subset.fq
	│   ├── Irrel_kd_3.subset.fq
	│   ├── Mov10_oe_1.subset.fq
	│   ├── Mov10_oe_2.subset.fq
	│   └── Mov10_oe_3.subset.fq
	├── results
	└── scripts
</code></pre></div></div>

<p>To use the STAR aligner, load the module:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>module load gcc/6.2.0 star/2.5.2b
</code></pre></div></div>

<p>Aligning reads using STAR is a two step process:</p>

<ol>
  <li>Create a genome index</li>
  <li>Map reads to the genome</li>
</ol>

<blockquote>
  <p>A quick note on shared databases for human and other commonly used model organisms. The O2 cluster has a designated directory at <code class="language-plaintext highlighter-rouge">/n/groups/shared_databases/</code> in which there are files that can be accessed by any user. These files contain, but are not limited to, genome indices for various tools, reference sequences, tool specific data, and data from public databases, such as NCBI and PDB. So when using a tool that requires a reference of sorts, it is worth taking a quick look here because chances are it’s already been taken care of for you.</p>

  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">ls</span> <span class="nt">-l</span> /n/groups/shared_databases/igenome/
</code></pre></div>  </div>
</blockquote>

<h3 id="creating-a-genome-index">Creating a genome index</h3>

<p>For this workshop we are using reads that originate from a small subsection of chromosome 1 (~300,000 reads) and so we are using only chr1 as the reference genome.</p>

<p>To store our genome indices, we will use the <code class="language-plaintext highlighter-rouge">/n/scratch2/</code> space with large temporary storage capacity. We need to create a directory for the indices within this space:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir</span> <span class="nt">-p</span> /n/scratch2/username/chr1_hg38_index
</code></pre></div></div>

<p>The basic options to <strong>generate genome indices</strong> using STAR are as follows:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--runThreadN</code>: number of threads</li>
  <li><code class="language-plaintext highlighter-rouge">--runMode</code>: genomeGenerate mode</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeDir</code>: /path/to/store/genome_indices</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeFastaFiles</code>: /path/to/FASTA_file</li>
  <li><code class="language-plaintext highlighter-rouge">--sjdbGTFfile</code>: /path/to/GTF_file</li>
  <li><code class="language-plaintext highlighter-rouge">--sjdbOverhang</code>: readlength -1</li>
</ul>

<blockquote>
  <p><em>NOTE:</em> In case of reads of varying length, the ideal value for <code class="language-plaintext highlighter-rouge">--sjdbOverhang</code> is max(ReadLength)-1. In most cases, the default value of 100 will work similarly to the ideal value.</p>
</blockquote>

<p>Now let’s create a job submission script to generate the genome index:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>vim ~/rnaseq/scripts/genome_index.run
</code></pre></div></div>
<p>Within <code class="language-plaintext highlighter-rouge">vim</code> we now add our shebang line, the SLURM directives, and our STAR command.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c">#!/bin/bash</span>

<span class="c">#SBATCH -p short 		# partition name</span>
<span class="c">#SBATCH -t 0-2:00 		# hours:minutes runlimit after which job will be killed</span>
<span class="c">#SBATCH -n 6 		# number of cores requested -- this needs to be greater than or equal to the number of cores you plan to use to run your job</span>
<span class="c">#SBATCH --mem 16G</span>
<span class="c">#SBATCH --job-name STAR_index 		# Job name</span>
<span class="c">#SBATCH -o %j.out			# File to which standard out will be written</span>
<span class="c">#SBATCH -e %j.err 		# File to which standard err will be written</span>

<span class="nb">cd</span> /n/scratch2/username/

module load gcc/6.2.0 star/2.5.2b

STAR <span class="nt">--runThreadN</span> 6 <span class="se">\</span>
<span class="nt">--runMode</span> genomeGenerate <span class="se">\</span>
<span class="nt">--genomeDir</span> chr1_hg38_index <span class="se">\</span>
<span class="nt">--genomeFastaFiles</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/Homo_sapiens.GRCh38.dna.chromosome.1.fa <span class="se">\</span>
<span class="nt">--sjdbGTFfile</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/Homo_sapiens.GRCh38.92.gtf <span class="se">\</span>
<span class="nt">--sjdbOverhang</span> 99
</code></pre></div></div>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>sbatch ~/rnaseq/scripts/genome_index.run
</code></pre></div></div>

<h3 id="aligning-reads">Aligning reads</h3>

<p>After you have the genome indices generated, you can perform the read alignment. We previously generated the genome indices for you in <code class="language-plaintext highlighter-rouge">/n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/ensembl38_STAR_index/</code> directory so that we don’t get held up waiting on the generation of the indices.</p>

<p>Create an output directory for our alignment files:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/rnaseq/raw_data

<span class="nv">$ </span><span class="nb">mkdir</span> ../results/STAR
</code></pre></div></div>

<h3 id="star-command-in-interactive-bash">STAR command in interactive bash</h3>

<p>For now, we’re going to work on just one sample to set up our workflow. To start we will use the first replicate in the Mov10 over-expression group, <code class="language-plaintext highlighter-rouge">Mov10_oe_1.subset.fq</code>. Details on STAR and its functionality can be found in the <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">user manual</a>; we encourage you to peruse through to get familiar with all available options.</p>

<p>The basic options for aligning reads to the genome using STAR are:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--runThreadN</code>: number of threads / cores</li>
  <li><code class="language-plaintext highlighter-rouge">--readFilesIn</code>: /path/to/FASTQ_file</li>
  <li><code class="language-plaintext highlighter-rouge">--genomeDir</code>: /path/to/genome_indices_directory</li>
  <li><code class="language-plaintext highlighter-rouge">--outFileNamePrefix</code>: prefix for all output files</li>
</ul>

<p>Listed below are additional parameters that we will use in our command:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">--outSAMtype</code>: output filetype (SAM default)</li>
  <li><code class="language-plaintext highlighter-rouge">--outSAMunmapped</code>: what to do with unmapped reads</li>
</ul>

<blockquote>
  <p><strong>NOTE:</strong> Default filtering is applied in which the maximum number of multiple alignments allowed for a read is set to 10. If a read exceeds this number there is no alignment output. To change the default you can use <code class="language-plaintext highlighter-rouge">--outFilterMultimapNmax</code>, but for this lesson we will leave it as default. Also, note that “<strong>STAR’s default parameters are optimized for mammalian genomes.</strong> Other species may require significant modifications of some alignment parameters; in particular, the maximum and minimum intron sizes have to be reduced for organisms with smaller introns” [<a href="http://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635.full.pdf+html">1</a>].</p>
</blockquote>

<p>We can access the software by simply using the STAR command followed by the basic parameters described above and any additional parameters. The full command is provided below for you to copy paste into your terminal. If you want to manually enter the command, it is advisable to first type out the full command in a text editor (i.e. <a href="http://www.sublimetext.com/">Sublime Text</a> or <a href="https://notepad-plus-plus.org/">Notepad++</a>) on your local machine and then copy paste into the terminal. This will make it easier to catch typos and make appropriate changes.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
STAR <span class="nt">--genomeDir</span> /n/groups/hbctraining/intro_rnaseq_hpc/reference_data_ensembl38/ensembl38_STAR_index/ <span class="se">\</span>
<span class="nt">--runThreadN</span> 6 <span class="se">\</span>
<span class="nt">--readFilesIn</span> Mov10_oe_1.subset.fq <span class="se">\</span>
<span class="nt">--outFileNamePrefix</span> ../results/STAR/Mov10_oe_1_ <span class="se">\</span>
<span class="nt">--outSAMtype</span> BAM SortedByCoordinate <span class="se">\</span>
<span class="nt">--outSAMunmapped</span> Within <span class="se">\</span>
<span class="nt">--outSAMattributes</span> Standard 

</code></pre></div></div>

<hr />
<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
