<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Working on HPC | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="Working on HPC" />
<meta name="author" content="Radhika Khetani, Meeta Mistry, Mary Piper, Jihe Liu" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/working_on_HPC_noExercises.html" />
<meta property="og:url" content="https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/working_on_HPC_noExercises.html" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2020-11-16T00:00:00+00:00" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Working on HPC" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"BlogPosting","author":{"@type":"Person","name":"Radhika Khetani, Meeta Mistry, Mary Piper, Jihe Liu"},"dateModified":"2020-11-16T00:00:00+00:00","datePublished":"2020-11-16T00:00:00+00:00","description":"Introduction to bulk RNA-seq","headline":"Working on HPC","mainEntityOfPage":{"@type":"WebPage","@id":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/working_on_HPC_noExercises.html"},"url":"https://esallychang.github.io/modified-intro-to-rnaseq-hpc/lessons/working_on_HPC_noExercises.html"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/modified-intro-to-rnaseq-hpc/assets/css/style.css?v=d838fbb944b8d2e4dc973fe8850414c54cfcdf99">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/modified-intro-to-rnaseq-hpc/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">Working on HPC</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <h1 id="working-in-an-hpc-environment">Working in an HPC environment</h1>

<h2 id="learning-objectives">Learning Objectives</h2>
<ul>
  <li>Define the basic components of a computing environment</li>
  <li>Explain parallelization in computing</li>
  <li>Deomnstrate logging in to the FAS-RC cluster and starting an interactive session</li>
  <li>Describe the salient features of the SLURM job scheduling system</li>
  <li>Describe the usage of the LMOD system</li>
  <li>Discuss storage options on the cluster</li>
</ul>

<h2 id="clarifying-some-terminology">Clarifying some terminology</h2>

<h3 id="cores">Cores</h3>
<p>CPUs (Central Processing Units) are composed of a single or multiple cores. If I have a processor with 4 “cores”, i.e. a “quad-core” processor. This means that my processor can do 4 distinct computations at the same time.</p>

<h3 id="data-storage">Data Storage</h3>
<p>Data storage is measured in bytes, usually Gigabytes (GB), Terabytes (TB), Petabytes (PB). My laptop has 1 TB of storage, which is pretty standard for laptops.</p>

<h3 id="memory">Memory</h3>
<p>Memory is a small amount of volatile or temporary information storage. This is distinct from data storage we discussed in the previous point. It is an essential component of any computer, as this is where data is stored <em>when some computation is being performed</em> on it. If you have ever used R, all the objects in your environment are usually stored in memory until you save your environment. My laptop has 16 GB of memory, which is a little higher than average.</p>

<h2 id="why-use-the-cluster-or-an-hpc-environment">Why use the cluster or an HPC environment?</h2>

<p align="center">
<img src="https://github.com/hbctraining/Intro-to-shell-flipped/blob/master/img/compute_cluster.png?raw=true" width="450" />
</p>

<ol>
  <li>A lot of software is designed to work with the resources on an HPC environment and is either unavailable for, or unusable on, a personal computer.</li>
  <li>If you are performing analysis on large data files (e.g. high-throughput sequencing data), you should work on the cluster to avoid issues with memory and to get the analysis done a lot faster with the superior processing capacity. Essentially, a cluster has:
    <ul>
      <li>100s of cores for processing!</li>
      <li>100s of Gigabytes or Petabytes of storage!</li>
      <li>100s of Gigabytes of memory!</li>
    </ul>
  </li>
</ol>

<h3 id="parallelization">Parallelization</h3>

<p>Point 2 in the last section brings us to the idea of <strong>parallelization</strong> or parallel computing that enables us to efficiently use the resources available on the cluster.</p>

<p>What if we had 3 input files that we wanted to analyze? Well, we could process these files <strong>in serial</strong>, i.e. use the same core(s) over and over again, with or without multithreading, as shown in the image below.</p>

<p align="center">
<img src="../img/serial_hpc_3samples.png" width="450" />
</p>

<p>This is great, but it is not as efficient as multithreading each analysis, and using a set of 8 cores for each of the three input samples. With this type of parallelization, several samples can be analysed at the same time!</p>

<p align="center">
<img src="../img/multithreaded_hpc_3samples.png" width="650" />
</p>

<h2 id="connecting-to-a-login-node-on-o2">Connecting to a <em>login</em> node on O2</h2>

<blockquote>
  <p><strong>NOTE: There is code provided below as an example, please do not try and run it.</strong></p>
</blockquote>

<p>The following command is used to log in to O2:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ssh username@o2.hms.harvard.edu
</code></pre></div></div>

<p><strong><code class="language-plaintext highlighter-rouge">ssh</code> stands for secure shell.</strong> All of the information (like your password) going between your computer and the O2 login computer is encrypted when using <code class="language-plaintext highlighter-rouge">ssh</code>.</p>

<p>Once logged in, you should see the O2 icon, some news, and the command prompt, e.g. <code class="language-plaintext highlighter-rouge">[rc_training10@login01 ~]$</code>.</p>

<p>About nodes:</p>
<ul>
  <li>A “node” on a cluster is essentially a computer in the cluster of computers.</li>
  <li>There are dedicated login nodes and compute nodes.</li>
  <li>A login node’s only function is to enable users to log in to a cluster, it is not meant to be used for any actual work/computing. There are 6 login nodes on O2.</li>
  <li>There are several hundred compute nodes on O2 available for performing your analysis/work.</li>
</ul>

<h2 id="connecting-to-a-compute-node-on-o2">Connecting to a <em>compute</em> node on O2</h2>

<p>You can access compute nodes in 2 ways using a job scheduler or resource manager like Slurm.</p>
<ol>
  <li>Directly using an interactive session (Slurm’s <code class="language-plaintext highlighter-rouge">srun</code> command):
    <ul>
      <li>The <code class="language-plaintext highlighter-rouge">srun</code> command with a few mandatory parameters will create an “interactive session” on O2.</li>
      <li>This is essentially a way for us to do work on the compute node directly from the terminal.</li>
      <li>If the connectivity to the cluster is lost in the middle of a command being run that work will be lost in an interactive session.</li>
    </ul>
  </li>
  <li>By starting a “batch” job (Slurm’s <code class="language-plaintext highlighter-rouge">sbatch</code> command):
    <ul>
      <li>The <code class="language-plaintext highlighter-rouge">sbatch</code> command with a few mandatory parameters + a specialized shell script will result in the script being run on a compute node.</li>
      <li>This “job” will not be accessible directly from the Terminal and will run in the background.</li>
      <li>Users do not need to remain connected to the cluster when such a “batch job” is running.</li>
    </ul>
  </li>
</ol>

<p>In this workshop, we will mostly create interactive sessions with <code class="language-plaintext highlighter-rouge">srun</code>.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Example code for starting an interactive session</span>
<span class="nv">$ </span>srun <span class="nt">--pty</span> <span class="nt">-p</span> interactive <span class="nt">-t</span> 0-3:00 <span class="nt">--mem</span> 1G <span class="nt">--reservation</span><span class="o">=</span>HBC1 /bin/bash
</code></pre></div></div>

<p>In the above command the parameters we are using are requesting specific resources:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">--pty</code> - Start an interactive session</li>
  <li><code class="language-plaintext highlighter-rouge">-p interactive</code> - on the “partition” called “interactive” (a partition is a group of computers dedicated to certain types of jobs, interactive, long, short, high-memory, etc.)</li>
  <li><code class="language-plaintext highlighter-rouge">-t 0-8:00</code> - time needed for this work: 0 days, 8 hours, 0 minutes.</li>
  <li><code class="language-plaintext highlighter-rouge">--mem 1G</code> - memory needed - 1 gibibyte (GiB)</li>
  <li><code class="language-plaintext highlighter-rouge">--reservation=HBC1</code> - <em>this is only for <strong>in class</strong> portions of this workshop, make sure you don’t use it for self-learning or when you have your own accounts.</em></li>
  <li><code class="language-plaintext highlighter-rouge">/bin/bash</code> - You want to interact with the compute node using the <em>bash</em> shell</li>
</ul>

<blockquote>
  <p>These parameters are used for <code class="language-plaintext highlighter-rouge">sbatch</code> as well, but they are listed differently within the script used to submit a batch job. We will be reviewing this later in this lesson.</p>
</blockquote>

<p><strong>After starting an interactive session, the command prompt now contains the word “compute”, <em>e.g. <code class="language-plaintext highlighter-rouge">[rc_training10@compute-a-16-163 ~]$</code></em>.</strong> 
You are now working on a compute node directly in an “interactive” session!</p>

<h2 id="more-about-slurm">More about Slurm</h2>

<ul>
  <li>Slurm = Simple Linux Utility for Resource Management</li>
  <li>Fairly allocates access to resources (computer nodes) to users for some duration of time so they can perform work</li>
  <li>Provides a framework for starting, executing, and monitoring batch jobs</li>
  <li>Manages a queue of pending jobs; ensures that no single user or core monopolizes the cluster</li>
</ul>

<h3 id="requesting-resources-from-slurm">Requesting resources from Slurm</h3>

<p>Below is table with some of the arguments you can specify when requesting resources from Slurm for both <code class="language-plaintext highlighter-rouge">srun</code> and <code class="language-plaintext highlighter-rouge">sbatch</code>:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Argument</th>
      <th style="text-align: center">Description / Input</th>
      <th style="text-align: center">Examples</th>
      <th style="text-align: center">Links</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-p</code></td>
      <td style="text-align: center">name of compute partition</td>
      <td style="text-align: center">short, medium, interactive</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/How+to+choose+a+partition+in+O2">O2 Wiki - Guidelines for choosing a partition</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-t</code></td>
      <td style="text-align: center">how much time to allocate to job</td>
      <td style="text-align: center">0-03:00, 5:00:00</td>
      <td style="text-align: center"><a href="https://harvardmed.atlassian.net/wiki/spaces/O2/pages/1586793632/Using+Slurm+Basic#Time-limits">O2 Wiki - Time limits for each partition</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-c</code></td>
      <td style="text-align: center">max cores</td>
      <td style="text-align: center">4, 8</td>
      <td style="text-align: center"><a href="https://harvardmed.atlassian.net/wiki/spaces/O2/pages/1586793632/Using+Slurm+Basic#How-many-cores?">O2 Wiki - How many cores?</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">--mem</code></td>
      <td style="text-align: center">max memory</td>
      <td style="text-align: center">8G, 8000</td>
      <td style="text-align: center"><a href="https://harvardmed.atlassian.net/wiki/spaces/O2/pages/1586793632/Using+Slurm+Basic#Memory-requirements">O2 Wiki - Memory requirements</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-o</code></td>
      <td style="text-align: center">name of file to create with standard output</td>
      <td style="text-align: center">%j.out</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/Using+Slurm+Basic#UsingSlurmBasic-sbatchoptionsquickreference">O2 Wiki</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-e</code></td>
      <td style="text-align: center">name of file to create with standard error</td>
      <td style="text-align: center">%j.err</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/Using+Slurm+Basic#UsingSlurmBasic-sbatchoptionsquickreference">O2 Wiki</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">-J</code></td>
      <td style="text-align: center">name of the job</td>
      <td style="text-align: center">Fastqc_run, rnaseq_workflow_mov10</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/Using+Slurm+Basic#UsingSlurmBasic-sbatchoptionsquickreference">O2 Wiki</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">--mail-type</code></td>
      <td style="text-align: center">send an email when job starts, ends or errors out</td>
      <td style="text-align: center">END, ALL</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/Using+Slurm+Basic#UsingSlurmBasic-sbatchoptionsquickreference">O2 Wiki</a></td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">--mail-user</code></td>
      <td style="text-align: center">send email to this address</td>
      <td style="text-align: center">xyz10@harvard.edu</td>
      <td style="text-align: center"><a href="https://wiki.rc.hms.harvard.edu/display/O2/Using+Slurm+Basic#UsingSlurmBasic-sbatchoptionsquickreference">O2 Wiki</a></td>
    </tr>
  </tbody>
</table>

<h3 id="sbatch-job-submission-script"><code class="language-plaintext highlighter-rouge">sbatch</code> job submission script</h3>

<p>An <code class="language-plaintext highlighter-rouge">sbatch</code> job submission script is essentially a normal shell script with the Slurm resource request specified at the top (Slurm directives) preceded by <code class="language-plaintext highlighter-rouge">#SBATCH</code>. Below is an example of an sbatch shell script that is requesting the following:</p>
<ul>
  <li>the “short” partition for 2 hours</li>
  <li>on 4 cores (30 minutes for each core)</li>
  <li>using 400MiB (100MiB for each core)</li>
</ul>

<p><strong><em>DO NOT RUN</em></strong></p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#! /bin/sh

#SBATCH -p short
#SBATCH –t 0-02:00
#SBATCH –c 4
#SBATCH --mem=400M
#SBATCH –o %j.out
#SBATCH –e %j.err
#SBATCH -J fastqc_run
#SBATCH --mail-type=ALL
#SBATCH –-mail-user=xyz10@med.harvard.edu

## Load the fastqc module
module load fastqc/0.11.5

# run fastqc (multithreaded)
fastqc -t 4 file1_1.fq file1_2.fq file2_1.fq file2_2.fq
</code></pre></div></div>

<h2 id="using-software-on-o2">Using software on O2</h2>

<h3 id="lmod-system">LMOD system</h3>

<p>In the above example we want to run the FastQC tool on four files. However, before we use the <code class="language-plaintext highlighter-rouge">fastqc</code> command, we’ve used the command <code class="language-plaintext highlighter-rouge">module load fastqc/0.11.5</code>. This <code class="language-plaintext highlighter-rouge">module load</code> command is part of the LMOD system available on O2. It enables users to access software installed on O2 easily, and manages every software’s dependency. The LMOD system adds directory paths of software executables and their dependencies (if any) into the <code class="language-plaintext highlighter-rouge">$PATH</code> variable.</p>

<p>So, instead of using <code class="language-plaintext highlighter-rouge">/n/app/fastqc/0.11.5/bin/fastqc</code> as our command, we can load the module and use <code class="language-plaintext highlighter-rouge">fastqc</code> as the command.</p>

<p>Some key LMOD commands are listed below:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">LMOD command</th>
      <th style="text-align: center">description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module spider</code></td>
      <td style="text-align: center">List all possible modules on the cluster</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module spider modulename</code></td>
      <td style="text-align: center">List all possible versions of that module</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module avail</code></td>
      <td style="text-align: center">List available modules available on the cluster</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module avail string</code></td>
      <td style="text-align: center">List available modules containing that string</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module load modulename/version</code></td>
      <td style="text-align: center">Add the full path to the tool to <code class="language-plaintext highlighter-rouge">$PATH</code> (and modify other environment variables)</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module list</code></td>
      <td style="text-align: center">List loaded modules</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module unload modulename/version</code></td>
      <td style="text-align: center">Unload a specific module</td>
    </tr>
    <tr>
      <td style="text-align: center"><code class="language-plaintext highlighter-rouge">module purge</code></td>
      <td style="text-align: center">Unload all loaded modules</td>
    </tr>
  </tbody>
</table>

<blockquote>
  <p>Note: On O2, a lot of tools used for analysis of sequencing data need to have the <code class="language-plaintext highlighter-rouge">gcc</code> compiler module loaded (<code class="language-plaintext highlighter-rouge">module load gcc/6.2.0</code>) prior to loading the tool of interest.</p>
</blockquote>

<h2 id="filesystems-on-o2">Filesystems on O2</h2>

<p align="center">
<img src="../img/O2_primary-storage.png" width="600" />
</p>

<ul>
  <li>Storage on HPC systems is organized differently than on your personal machine.</li>
  <li>Each node on the cluster does not have storage; instead, it is on disks bundled together externally.</li>
  <li>Storage filesystems can be quite complex, with large spaces dedicated to a pre-defined purpose.</li>
  <li>Filesystems are accessed over the internal network by all the nodes on the cluster.</li>
  <li>There are 3 major groups on the O2 cluster, each with their features and constraints:
    <ol>
      <li><code class="language-plaintext highlighter-rouge">/n/data1</code>, <code class="language-plaintext highlighter-rouge">/n/data2</code>, <code class="language-plaintext highlighter-rouge">/n/groups</code> - Large datasets are stored in these parent directories (see features/constraints in the image above).</li>
      <li><code class="language-plaintext highlighter-rouge">/home</code> - the home directories of all users are under this parent directory (see features/constraints in the image above).</li>
      <li><code class="language-plaintext highlighter-rouge">/n/scratch</code> - scratch space for temporary storage.</li>
    </ol>
  </li>
</ul>

<p><strong><a href="https://it.hms.harvard.edu/our-services/research-computing/services/storage">Please find more information about storage on O2 by clicking here.</a></strong></p>

<h3 id="more-about-nscratch">More about <code class="language-plaintext highlighter-rouge">/n/scratch</code></h3>

<ul>
  <li>It is for data only needed temporarily during analyses.</li>
  <li>Each user can use up to 25 TB and 2.5 million files/directories.</li>
  <li>Files not accessed for 45 days are automatically deleted.</li>
  <li><strong>No backups!</strong></li>
  <li>You can create your own folder using this command <code class="language-plaintext highlighter-rouge">/n/cluster/bin/scratch_create_directory.sh</code></li>
</ul>

<hr />
<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
