<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">

<!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Workshop Schedule | GitHub Pages Test</title>
<meta name="generator" content="Jekyll v3.10.0" />
<meta property="og:title" content="Workshop Schedule" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Introduction to bulk RNA-seq" />
<meta property="og:description" content="Introduction to bulk RNA-seq" />
<link rel="canonical" href="https://esallychang.github.io/branched-docs-demo/modified-intro-apr2025/schedule/" />
<meta property="og:url" content="https://esallychang.github.io/branched-docs-demo/modified-intro-apr2025/schedule/" />
<meta property="og:site_name" content="GitHub Pages Test" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Workshop Schedule" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","description":"Introduction to bulk RNA-seq","headline":"Workshop Schedule","url":"https://esallychang.github.io/branched-docs-demo/modified-intro-apr2025/schedule/"}</script>
<!-- End Jekyll SEO tag -->

    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:400,700&display=swap" as="style" type="text/css" crossorigin>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="theme-color" content="#157878">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <link rel="stylesheet" href="/branched-docs-demo/modified-intro-apr2025/assets/css/style.css?v=55b8125905b1663ec73ee9c29b169c4ffd03076a">
    <!-- start custom head snippets, customize with your own _includes/head-custom.html file -->

<!-- Setup Google Analytics -->



<!-- You can set your favicon here -->
<!-- link rel="shortcut icon" type="image/x-icon" href="/branched-docs-demo/modified-intro-apr2025/favicon.ico" -->

<!-- end custom head snippets -->

  </head>
  <body>
    <a id="skip-to-content" href="#content">Skip to the content.</a>

    <header class="page-header" role="banner">
      <h1 class="project-name">Workshop Schedule</h1>
      <h2 class="project-tagline">Introduction to bulk RNA-seq</h2>
      
        <a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc" class="btn">View on GitHub</a>
      
      
    </header>

    <main id="content" class="main-content" role="main">
      <h1 id="workshop-schedule">Workshop Schedule</h1>

<blockquote>
  <p><strong>Pre-requisite for this workshop:</strong> The <em>Basic Data Skills</em> <a href="https://hbctraining.github.io/Intro-to-shell-flipped/schedule/">Introduction to the command-line interface</a> workshop or a working knowledge of the command line and cluster computing.</p>
</blockquote>

<h2 id="pre-reading">Pre-reading</h2>

<ul>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/shell_review.html">Shell basics review</a></li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/04a_data_organization.html">Best Practices in Research Data Management (RDM)</a></li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/01_intro-to-RNAseq.html">Introduction to RNA-seq</a></li>
</ul>

<h2 id="day-1">Day 1</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Time</th>
      <th style="text-align: center">Topic</th>
      <th style="text-align: center">Instructor</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">09:30 - 09:45</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lectures/Intro_to_workshop.pdf">Workshop Introduction</a></td>
      <td style="text-align: center">Meeta</td>
    </tr>
    <tr>
      <td style="text-align: center">09:45 - 10:25</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lessons/03_working_on_HPC.html">Working in an HPC environment - Review</a></td>
      <td style="text-align: center">Emma</td>
    </tr>
    <tr>
      <td style="text-align: center">10:25 - 11:05</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lessons/04b_data_organization.html">Project Organization (using Data Management best practices)</a></td>
      <td style="text-align: center">Meeta</td>
    </tr>
    <tr>
      <td style="text-align: center">11:05 - 11:45</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lessons/05_qc_running_fastqc_interactively.html">Quality Control of Sequence Data: Running FASTQC</a></td>
      <td style="text-align: center">Emma</td>
    </tr>
    <tr>
      <td style="text-align: center">11:45 - 12:00</td>
      <td style="text-align: center">Overview of self-learning materials and homework submission</td>
      <td style="text-align: center">Meeta</td>
    </tr>
  </tbody>
</table>

<h3 id="before-the-next-class">Before the next class:</h3>

<ol>
  <li>Please <strong>study the contents</strong> and <strong>work through all the code</strong> within the following lessons:</li>
</ol>

<ul>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/02_experimental_planning_considerations.html">Experimental design considerations</a></li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/06_qc_running_fastqc_sbatch.html">Quality Control of Sequence Data: Running FASTQC on multiple samples</a></li>
  <li>
    <p><a href="/branched-docs-demo/modified-intro-apr2025/lessons/07_qc_fastqc_assessment.html">Quality Control of Sequence Data: Evaluating FASTQC reports</a></p>

    <blockquote>
      <p><strong>NOTE:</strong> To run through the code above, you will need to be <strong>logged into O2</strong> and <strong>working on a compute node</strong> (i.e. your command prompt should have the word <code class="language-plaintext highlighter-rouge">compute</code> in it).</p>
      <ol>
        <li>Log in using <code class="language-plaintext highlighter-rouge">ssh rc_trainingXX@o2.hms.harvard.edu</code> and enter your password (replace the “XX” in the username with the number you were <a href="https://docs.google.com/spreadsheets/d/1kBlYowhjjHJC9ZovmbBULmbqozKpprM17vZ2wPlhNg0/edit#gid=0">assigned in class</a>).</li>
        <li>Once you are on the login node, use <code class="language-plaintext highlighter-rouge">srun --pty -p interactive -t 0-2:30 --mem 1G /bin/bash</code> to get on a compute node or as specified in the lesson.</li>
        <li>Proceed only once your command prompt has the word <code class="language-plaintext highlighter-rouge">compute</code> in it.</li>
        <li>If you log out between lessons (using the <code class="language-plaintext highlighter-rouge">exit</code> command twice), please follow points 1. and 2. above to log back in and get on a compute node when you restart with the self learning.</li>
      </ol>
    </blockquote>
  </li>
</ul>

<ol>
  <li><strong>Complete the exercises</strong>:
    <ul>
      <li>Each lesson above contain exercises; please go through each of them.</li>
      <li>Add your answers to the questions to <a href="https://docs.google.com/forms/d/e/1FAIpQLSdxdSM4528uYTWT7k5c8gYAuCUaTqRkUSI88eUmKg7qyQZZAQ/viewform?usp=sf_link">Google forms</a> the <strong>day before the next class</strong>.</li>
    </ul>
  </li>
</ol>

<h3 id="questions">Questions?</h3>
<ul>
  <li><strong><em>If you get stuck due to an error</em></strong> while runnning code in the lesson, <a href="mailto:hbctraining@hsph.harvard.edu">email us</a></li>
</ul>

<hr />

<h2 id="day-2">Day 2</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Time</th>
      <th style="text-align: center">Topic</th>
      <th style="text-align: center">Instructor</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">09:30 - 10:30</td>
      <td style="text-align: center">Self-learning lessons review</td>
      <td style="text-align: center">All</td>
    </tr>
    <tr>
      <td style="text-align: center">10:30 - 11:10</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lectures/expression_quantification.pdf">Expression quantification: Theory and Tools</a></td>
      <td style="text-align: center">Meeta</td>
    </tr>
    <tr>
      <td style="text-align: center">11:10 - 11:50</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lessons/08_quasi_alignment_salmon.html">Quantifying expression using alignment-free methods (Salmon)</a></td>
      <td style="text-align: center">Emma</td>
    </tr>
    <tr>
      <td style="text-align: center">11:50 - 12:00</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lectures/workflow_overview.pdf">Review of workflow</a></td>
      <td style="text-align: center">Emma</td>
    </tr>
  </tbody>
</table>

<h3 id="before-the-next-class-1">Before the next class:</h3>

<ol>
  <li>Please <strong>study the contents</strong> and <strong>work through all the code</strong> within the following lessons:</li>
</ol>

<ul>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/09_quasi_alignment_salmon_sbatch.html">Quantifying expression using alignment-free methods (Salmon on multiple samples)</a>
    <details>
    <summary><i>Click here for a preview of this lesson</i></summary>
      <br />Now that we know how to run the quantification of one sample with Salmon, this lesson will guide you to run multiple samples by creating a job submission script<br /><br />
    </details>
  </li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/10_QC_Qualimap.html">QC with Alignment Data</a>
    <details>
    <summary><i>Click here for a preview of this lesson</i></summary>
      <br />Besides transcript-level quantification, we also want to understand the quality of the mapping, which is not provided in Salmon output. <br /><br />This lesson will cover:<br />
          - Aligning the reads with an aligner, STAR<br />
          - Assessing QC metrics among samples<br /><br />
    </details>
  </li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/lessons/11_multiQC.html">Documenting Steps in the Workflow with MultiQC</a>
    <details>
    <summary><i>Click here for a preview of this lesson</i></summary>
      <br />It would be great to have a summary document of all QC results from the previous analysis. <br /><br />This lesson will cover:<br />
          - Generating such a summary report with multiQC<br />
          - Generating alignment metric with Qualimap<br /><br />
    </details>

    <blockquote>
      <p><strong>NOTE:</strong> To run through the code above, you will need to be <strong>logged into O2</strong> and <strong>working on a compute node</strong> (i.e. your command prompt should have the word <code class="language-plaintext highlighter-rouge">compute</code> in it).</p>
      <ol>
        <li>Log in using <code class="language-plaintext highlighter-rouge">ssh rc_trainingXX@o2.hms.harvard.edu</code> and enter your password (replace the “XX” in the username with the number you were assigned in class).</li>
        <li>Once you are on the login node, use <code class="language-plaintext highlighter-rouge">srun --pty -p interactive -t 0-2:30 --mem 8G /bin/bash</code> to get on a compute node or as specified in the lesson.</li>
        <li>Proceed only once your command prompt has the word <code class="language-plaintext highlighter-rouge">compute</code> in it.</li>
        <li>If you log out between lessons (using the <code class="language-plaintext highlighter-rouge">exit</code> command twice), please follow points 1. and 2. above to log back in and get on a compute node when you restart with the self learning.</li>
      </ol>
    </blockquote>
  </li>
</ul>

<ol>
  <li><strong>Complete the exercises</strong>:
    <ul>
      <li>Each lesson above contain exercises; please go through each of them.</li>
      <li>Add your answers to the questions to <a href="https://docs.google.com/forms/d/e/1FAIpQLScxaj3IIO4Bx7FCRw87cCeuTPQyhD_7WR2QU638y8IZDv5r1A/viewform?usp=sf_link">Google forms</a> the <strong>day before the next class</strong>.</li>
    </ul>
  </li>
</ol>

<h3 id="questions-1">Questions?</h3>
<ul>
  <li><strong><em>If you get stuck due to an error</em></strong> while runnning code in the lesson, <a href="mailto:hbctraining@hsph.harvard.edu">email us</a></li>
</ul>

<hr />

<h2 id="day-3">Day 3</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Time</th>
      <th style="text-align: center">Topic</th>
      <th style="text-align: center">Instructor</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">09:30 - 10:10</td>
      <td style="text-align: center">Self-learning lessons review</td>
      <td style="text-align: center">All</td>
    </tr>
    <tr>
      <td style="text-align: center">10:10 - 11:10</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lessons/12_automating_workflow.html">Automating the RNA-seq workflow</a></td>
      <td style="text-align: center">Will</td>
    </tr>
    <tr>
      <td style="text-align: center">11:10 - 11:45</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lectures/RNA-seq_troubleshooting.pdf">Troubleshooting RNA-seq Data Analysis</a></td>
      <td style="text-align: center">Emma</td>
    </tr>
    <tr>
      <td style="text-align: center">11:45 - 12:00</td>
      <td style="text-align: center"><a href="/branched-docs-demo/modified-intro-apr2025/lectures/workshop_wrapup.pdf">Wrap up</a></td>
      <td style="text-align: center">Will</td>
    </tr>
  </tbody>
</table>

<hr />

<ul>
  <li>Downloadable Answer Keys (Day 2 exercises):
    <ul>
      <li><a href="https://www.dropbox.com/s/524mevuyba34l5b/exp_design_table.xlsx?dl=1">Experimental design (one possible solution)</a></li>
      <li><a href="https://www.dropbox.com/s/9wdyhfqpic05l6p/mov10_fastqc.run?dl=1">sbatch script</a></li>
      <li><a href="https://www.dropbox.com/s/l7puf8oahtbwmpk/22914006.out?dl=1">.out file</a></li>
      <li><a href="https://www.dropbox.com/s/8a1g6o9t2kxit30/22914006.err?dl=1">.err file</a>.</li>
    </ul>
  </li>
  <li>Downloadable Answer Keys (Day 3 exercises):
    <ul>
      <li><a href="/branched-docs-demo/modified-intro-apr2025/answer_key/salmon_all_samples.sbatch">sbatch script to run salmon for all samples</a></li>
    </ul>
  </li>
  <li><a href="/branched-docs-demo/modified-intro-apr2025/scripts/rnaseq_analysis_on_input_file.sh">Automation Script</a></li>
</ul>

<hr />

<h2 id="resources">Resources</h2>
<ul>
  <li><a href="https://harvardmed.atlassian.net/wiki/spaces/O2/pages/1918304257/How+to+request+an+O2+account">Getting an O2 account</a></li>
  <li><a href="https://www.youtube.com/watch?v=TMLIxwDP7sk">Video about statistics behind salmon quantification</a></li>
  <li>Advanced bash for working on O2:
    <ul>
      <li><a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionVI/lessons/more_bash.html#alias">Creating shortcuts or aliases</a></li>
      <li><a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionVI/lessons/more_bash.html#rsync">Copying files from other remote locations to O2</a></li>
      <li><a href="https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionVI/lessons/more_bash.html#symlink">Creating symbolic links</a></li>
    </ul>
  </li>
  <li><a href="https://hbctraining.github.io/Accessing_public_genomic_data/lessons/accessing_genome_reference_data.html">Obtaining reference genomes or transcriptomes</a></li>
  <li>Youtube videos
    <ul>
      <li><a href="https://www.youtube.com/watch?v=MfhjkfocRR0&amp;ab_channel=PaulProgramming">Hash tables - Paul Programming</a></li>
      <li><a href="https://www.youtube.com/watch?v=zqKlL3ZpTqs">Suffix arrays - William Fiset</a>
***</li>
    </ul>
  </li>
</ul>

<h2 id="building-on-this-workshop">Building on this workshop</h2>
<ul>
  <li><a href="https://hbctraining.github.io/Intro-to-R-flipped/#lessons">Introduction to R workshop materials</a></li>
  <li><a href="https://hbctraining.github.io/DGE_workshop_salmon_online/#lessons">Introduction to Differential Gene Expression analysis (bulk RNA-seq) workshop materials</a></li>
</ul>

<hr />
<p><em>These materials have been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


      <footer class="site-footer">
        
          <span class="site-footer-owner"><a href="https://github.com/esallychang/modified-intro-to-rnaseq-hpc">modified-intro-to-rnaseq-hpc</a> is maintained by <a href="https://github.com/esallychang">esallychang</a>.</span>
        
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a>.</span>
      </footer>
    </main>
  </body>
</html>
